// @cosmology-ui/utils/main/utils/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WalletStatus = exports.WalletMode = exports.Themes = exports.QRCodeStatus = exports.LogoStatus = exports.ButtonShape = void 0;
/* ====================================================== */
/*                      default type                      */
/* ====================================================== */
var WalletStatus;
exports.WalletStatus = WalletStatus;
(function (WalletStatus) {
  WalletStatus["Disconnected"] = "Disconnected";
  WalletStatus["Connecting"] = "Connecting";
  WalletStatus["Connected"] = "Connected";
  WalletStatus["NotExist"] = "NotExist";
  WalletStatus["Rejected"] = "Rejected";
  WalletStatus["Error"] = "Error";
})(WalletStatus || (exports.WalletStatus = WalletStatus = {}));
var Themes;
exports.Themes = Themes;
(function (Themes) {
  Themes["Light"] = "light";
  Themes["Dark"] = "dark";
})(Themes || (exports.Themes = Themes = {}));
/* ====================================================== */
/*                  Simple Connect Modal                  */
/* ====================================================== */
var LogoStatus;
exports.LogoStatus = LogoStatus;
(function (LogoStatus) {
  LogoStatus["Loading"] = "loading";
  LogoStatus["Warning"] = "warning";
  LogoStatus["Error"] = "error";
})(LogoStatus || (exports.LogoStatus = LogoStatus = {}));
var WalletMode;
exports.WalletMode = WalletMode;
(function (WalletMode) {
  WalletMode["Extension"] = "extension";
  WalletMode["WalletConnect"] = "wallet-connect";
})(WalletMode || (exports.WalletMode = WalletMode = {}));
var ButtonShape;
exports.ButtonShape = ButtonShape;
(function (ButtonShape) {
  ButtonShape["Square"] = "Square";
  ButtonShape["Rectangle"] = "Rectangle";
})(ButtonShape || (exports.ButtonShape = ButtonShape = {}));
var QRCodeStatus;
exports.QRCodeStatus = QRCodeStatus;
(function (QRCodeStatus) {
  QRCodeStatus["Pending"] = "Pending";
  QRCodeStatus["Done"] = "Done";
  QRCodeStatus["Error"] = "Error";
  QRCodeStatus["Expired"] = "Expired";
})(QRCodeStatus || (exports.QRCodeStatus = QRCodeStatus = {}));
