// @cosmology-ui/utils/module/cosmos/modals/SimpleModalHead.js
import { IconButton, Stack, Text } from '@chakra-ui/react';
import React, { useContext } from 'react';
import { FiChevronLeft, FiX } from 'react-icons/fi';
import { ThemeContext } from '../../index';
export const SimpleModalHeadBaseStyle = (theme, backButton) => {
  return {
    w: 'full',
    alignItems: 'center',
    h: 'fit-content',
    mb: 1,
    p: 4,
    pb: 1.5,
    '>.modal-header-text': {
      flex: 1,
      mr: backButton ? 0 : -10,
      fontSize: 'md',
      fontWeight: 'semibold',
      textAlign: 'center',
      color: `simple-modal-head-text-color-${theme}`
    },
    '>.icon-button': {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 'md',
      color: `simple-modal-head-button-icon-color-${theme}`,
      minW: 8,
      minH: 8,
      maxW: 8,
      maxH: 8,
      h: 8,
      w: 8,
      p: 0,
      _focus: {
        outline: 'none'
      },
      '>svg': {
        w: 5,
        h: 5
      }
    }
  };
};
export const SimpleModalHead = ({
  title,
  backButton,
  className,
  styleProps,
  onBack,
  onClose
}) => {
  const {
    theme
  } = useContext(ThemeContext);
  return /*#__PURE__*/React.createElement(Stack, {
    isInline: true,
    className: className,
    sx: styleProps ? styleProps : SimpleModalHeadBaseStyle(theme, backButton ? true : false)
  }, backButton ? /*#__PURE__*/React.createElement(IconButton, {
    "aria-label": "back",
    className: "icon-button",
    variant: "ghost",
    icon: /*#__PURE__*/React.createElement(FiChevronLeft, null),
    onClick: onBack
  }) : undefined, /*#__PURE__*/React.createElement(Text, {
    className: "modal-header-text"
  }, title), /*#__PURE__*/React.createElement(IconButton, {
    "aria-label": "close",
    className: "icon-button",
    variant: "ghost",
    icon: /*#__PURE__*/React.createElement(FiX, null),
    onClick: onClose
  }));
};
