// stargazejs/main/codegen/BaseMinter.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BaseMinterClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var BaseMinterClient = /*#__PURE__*/(0, _createClass2["default"])(function BaseMinterClient(client, sender, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, BaseMinterClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "sender", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "mint", /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(_ref) {
      var tokenUri,
        fee,
        memo,
        _funds,
        _args = arguments;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              tokenUri = _ref.tokenUri;
              fee = _args.length > 1 && _args[1] !== undefined ? _args[1] : "auto";
              memo = _args.length > 2 ? _args[2] : undefined;
              _funds = _args.length > 3 ? _args[3] : undefined;
              _context.next = 6;
              return _this.client.execute(_this.sender, _this.contractAddress, {
                mint: {
                  token_uri: tokenUri
                }
              }, fee, memo, _funds);
            case 6:
              return _context.abrupt("return", _context.sent);
            case 7:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "updateStartTradingTime", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
    var fee,
      memo,
      _funds,
      _args2 = arguments;
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            fee = _args2.length > 0 && _args2[0] !== undefined ? _args2[0] : "auto";
            memo = _args2.length > 1 ? _args2[1] : undefined;
            _funds = _args2.length > 2 ? _args2[2] : undefined;
            _context2.next = 5;
            return _this.client.execute(_this.sender, _this.contractAddress, {
              update_start_trading_time: {}
            }, fee, memo, _funds);
          case 5:
            return _context2.abrupt("return", _context2.sent);
          case 6:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  })));
  this.client = client;
  this.sender = sender;
  this.contractAddress = contractAddress;
  this.mint = this.mint.bind(this);
  this.updateStartTradingTime = this.updateStartTradingTime.bind(this);
});
exports.BaseMinterClient = BaseMinterClient;
