// stargazejs/main/codegen/BaseFactory.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BaseFactoryClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var BaseFactoryClient = /*#__PURE__*/(0, _createClass2["default"])(function BaseFactoryClient(client, sender, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, BaseFactoryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "sender", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "createMinter", /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(_ref) {
      var collectionParams,
        initMsg,
        fee,
        memo,
        _funds,
        _args = arguments;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              collectionParams = _ref.collectionParams, initMsg = _ref.initMsg;
              fee = _args.length > 1 && _args[1] !== undefined ? _args[1] : "auto";
              memo = _args.length > 2 ? _args[2] : undefined;
              _funds = _args.length > 3 ? _args[3] : undefined;
              _context.next = 6;
              return _this.client.execute(_this.sender, _this.contractAddress, {
                create_minter: {
                  collection_params: collectionParams,
                  init_msg: initMsg
                }
              }, fee, memo, _funds);
            case 6:
              return _context.abrupt("return", _context.sent);
            case 7:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }());
  this.client = client;
  this.sender = sender;
  this.contractAddress = contractAddress;
  this.createMinter = this.createMinter.bind(this);
});
exports.BaseFactoryClient = BaseFactoryClient;
