// stargazejs/main/codegen/Marketplace.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MarketplaceQueryClient = exports.MarketplaceClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var MarketplaceQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function MarketplaceQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, MarketplaceQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "collections", /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(_ref) {
      var limit, startAfter;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              limit = _ref.limit, startAfter = _ref.startAfter;
              return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                collections: {
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "ask", /*#__PURE__*/function () {
    var _ref4 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(_ref3) {
      var collection, tokenId;
      return _regenerator["default"].wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              collection = _ref3.collection, tokenId = _ref3.tokenId;
              return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                ask: {
                  collection: collection,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }));
    return function (_x2) {
      return _ref4.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "asks", /*#__PURE__*/function () {
    var _ref6 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(_ref5) {
      var collection, includeInactive, limit, startAfter;
      return _regenerator["default"].wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              collection = _ref5.collection, includeInactive = _ref5.includeInactive, limit = _ref5.limit, startAfter = _ref5.startAfter;
              return _context3.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                asks: {
                  collection: collection,
                  include_inactive: includeInactive,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3);
    }));
    return function (_x3) {
      return _ref6.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "reverseAsks", /*#__PURE__*/function () {
    var _ref8 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(_ref7) {
      var collection, includeInactive, limit, startBefore;
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              collection = _ref7.collection, includeInactive = _ref7.includeInactive, limit = _ref7.limit, startBefore = _ref7.startBefore;
              return _context4.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                reverse_asks: {
                  collection: collection,
                  include_inactive: includeInactive,
                  limit: limit,
                  start_before: startBefore
                }
              }));
            case 2:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4);
    }));
    return function (_x4) {
      return _ref8.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "asksSortedByPrice", /*#__PURE__*/function () {
    var _ref10 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(_ref9) {
      var collection, includeInactive, limit, startAfter;
      return _regenerator["default"].wrap(function _callee5$(_context5) {
        while (1) {
          switch (_context5.prev = _context5.next) {
            case 0:
              collection = _ref9.collection, includeInactive = _ref9.includeInactive, limit = _ref9.limit, startAfter = _ref9.startAfter;
              return _context5.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                asks_sorted_by_price: {
                  collection: collection,
                  include_inactive: includeInactive,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context5.stop();
          }
        }
      }, _callee5);
    }));
    return function (_x5) {
      return _ref10.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "reverseAsksSortedByPrice", /*#__PURE__*/function () {
    var _ref12 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6(_ref11) {
      var collection, includeInactive, limit, startBefore;
      return _regenerator["default"].wrap(function _callee6$(_context6) {
        while (1) {
          switch (_context6.prev = _context6.next) {
            case 0:
              collection = _ref11.collection, includeInactive = _ref11.includeInactive, limit = _ref11.limit, startBefore = _ref11.startBefore;
              return _context6.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                reverse_asks_sorted_by_price: {
                  collection: collection,
                  include_inactive: includeInactive,
                  limit: limit,
                  start_before: startBefore
                }
              }));
            case 2:
            case "end":
              return _context6.stop();
          }
        }
      }, _callee6);
    }));
    return function (_x6) {
      return _ref12.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "askCount", /*#__PURE__*/function () {
    var _ref14 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7(_ref13) {
      var collection;
      return _regenerator["default"].wrap(function _callee7$(_context7) {
        while (1) {
          switch (_context7.prev = _context7.next) {
            case 0:
              collection = _ref13.collection;
              return _context7.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                ask_count: {
                  collection: collection
                }
              }));
            case 2:
            case "end":
              return _context7.stop();
          }
        }
      }, _callee7);
    }));
    return function (_x7) {
      return _ref14.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "asksBySeller", /*#__PURE__*/function () {
    var _ref16 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee8(_ref15) {
      var includeInactive, limit, seller, startAfter;
      return _regenerator["default"].wrap(function _callee8$(_context8) {
        while (1) {
          switch (_context8.prev = _context8.next) {
            case 0:
              includeInactive = _ref15.includeInactive, limit = _ref15.limit, seller = _ref15.seller, startAfter = _ref15.startAfter;
              return _context8.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                asks_by_seller: {
                  include_inactive: includeInactive,
                  limit: limit,
                  seller: seller,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context8.stop();
          }
        }
      }, _callee8);
    }));
    return function (_x8) {
      return _ref16.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "bid", /*#__PURE__*/function () {
    var _ref18 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee9(_ref17) {
      var bidder, collection, tokenId;
      return _regenerator["default"].wrap(function _callee9$(_context9) {
        while (1) {
          switch (_context9.prev = _context9.next) {
            case 0:
              bidder = _ref17.bidder, collection = _ref17.collection, tokenId = _ref17.tokenId;
              return _context9.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                bid: {
                  bidder: bidder,
                  collection: collection,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context9.stop();
          }
        }
      }, _callee9);
    }));
    return function (_x9) {
      return _ref18.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "bidsByBidder", /*#__PURE__*/function () {
    var _ref20 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee10(_ref19) {
      var bidder, limit, startAfter;
      return _regenerator["default"].wrap(function _callee10$(_context10) {
        while (1) {
          switch (_context10.prev = _context10.next) {
            case 0:
              bidder = _ref19.bidder, limit = _ref19.limit, startAfter = _ref19.startAfter;
              return _context10.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                bids_by_bidder: {
                  bidder: bidder,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context10.stop();
          }
        }
      }, _callee10);
    }));
    return function (_x10) {
      return _ref20.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "bidsByBidderSortedByExpiration", /*#__PURE__*/function () {
    var _ref22 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee11(_ref21) {
      var bidder, limit, startAfter;
      return _regenerator["default"].wrap(function _callee11$(_context11) {
        while (1) {
          switch (_context11.prev = _context11.next) {
            case 0:
              bidder = _ref21.bidder, limit = _ref21.limit, startAfter = _ref21.startAfter;
              return _context11.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                bids_by_bidder_sorted_by_expiration: {
                  bidder: bidder,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context11.stop();
          }
        }
      }, _callee11);
    }));
    return function (_x11) {
      return _ref22.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "bids", /*#__PURE__*/function () {
    var _ref24 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee12(_ref23) {
      var collection, limit, startAfter, tokenId;
      return _regenerator["default"].wrap(function _callee12$(_context12) {
        while (1) {
          switch (_context12.prev = _context12.next) {
            case 0:
              collection = _ref23.collection, limit = _ref23.limit, startAfter = _ref23.startAfter, tokenId = _ref23.tokenId;
              return _context12.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                bids: {
                  collection: collection,
                  limit: limit,
                  start_after: startAfter,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context12.stop();
          }
        }
      }, _callee12);
    }));
    return function (_x12) {
      return _ref24.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "bidsSortedByPrice", /*#__PURE__*/function () {
    var _ref26 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee13(_ref25) {
      var collection, limit, startAfter;
      return _regenerator["default"].wrap(function _callee13$(_context13) {
        while (1) {
          switch (_context13.prev = _context13.next) {
            case 0:
              collection = _ref25.collection, limit = _ref25.limit, startAfter = _ref25.startAfter;
              return _context13.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                bids_sorted_by_price: {
                  collection: collection,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context13.stop();
          }
        }
      }, _callee13);
    }));
    return function (_x13) {
      return _ref26.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "reverseBidsSortedByPrice", /*#__PURE__*/function () {
    var _ref28 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee14(_ref27) {
      var collection, limit, startBefore;
      return _regenerator["default"].wrap(function _callee14$(_context14) {
        while (1) {
          switch (_context14.prev = _context14.next) {
            case 0:
              collection = _ref27.collection, limit = _ref27.limit, startBefore = _ref27.startBefore;
              return _context14.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                reverse_bids_sorted_by_price: {
                  collection: collection,
                  limit: limit,
                  start_before: startBefore
                }
              }));
            case 2:
            case "end":
              return _context14.stop();
          }
        }
      }, _callee14);
    }));
    return function (_x14) {
      return _ref28.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "collectionBid", /*#__PURE__*/function () {
    var _ref30 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee15(_ref29) {
      var bidder, collection;
      return _regenerator["default"].wrap(function _callee15$(_context15) {
        while (1) {
          switch (_context15.prev = _context15.next) {
            case 0:
              bidder = _ref29.bidder, collection = _ref29.collection;
              return _context15.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                collection_bid: {
                  bidder: bidder,
                  collection: collection
                }
              }));
            case 2:
            case "end":
              return _context15.stop();
          }
        }
      }, _callee15);
    }));
    return function (_x15) {
      return _ref30.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "collectionBidsByBidder", /*#__PURE__*/function () {
    var _ref32 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee16(_ref31) {
      var bidder, limit, startAfter;
      return _regenerator["default"].wrap(function _callee16$(_context16) {
        while (1) {
          switch (_context16.prev = _context16.next) {
            case 0:
              bidder = _ref31.bidder, limit = _ref31.limit, startAfter = _ref31.startAfter;
              return _context16.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                collection_bids_by_bidder: {
                  bidder: bidder,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context16.stop();
          }
        }
      }, _callee16);
    }));
    return function (_x16) {
      return _ref32.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "collectionBidsByBidderSortedByExpiration", /*#__PURE__*/function () {
    var _ref34 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee17(_ref33) {
      var bidder, limit, startAfter;
      return _regenerator["default"].wrap(function _callee17$(_context17) {
        while (1) {
          switch (_context17.prev = _context17.next) {
            case 0:
              bidder = _ref33.bidder, limit = _ref33.limit, startAfter = _ref33.startAfter;
              return _context17.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                collection_bids_by_bidder_sorted_by_expiration: {
                  bidder: bidder,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context17.stop();
          }
        }
      }, _callee17);
    }));
    return function (_x17) {
      return _ref34.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "collectionBidsSortedByPrice", /*#__PURE__*/function () {
    var _ref36 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee18(_ref35) {
      var collection, limit, startAfter;
      return _regenerator["default"].wrap(function _callee18$(_context18) {
        while (1) {
          switch (_context18.prev = _context18.next) {
            case 0:
              collection = _ref35.collection, limit = _ref35.limit, startAfter = _ref35.startAfter;
              return _context18.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                collection_bids_sorted_by_price: {
                  collection: collection,
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context18.stop();
          }
        }
      }, _callee18);
    }));
    return function (_x18) {
      return _ref36.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "reverseCollectionBidsSortedByPrice", /*#__PURE__*/function () {
    var _ref38 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee19(_ref37) {
      var collection, limit, startBefore;
      return _regenerator["default"].wrap(function _callee19$(_context19) {
        while (1) {
          switch (_context19.prev = _context19.next) {
            case 0:
              collection = _ref37.collection, limit = _ref37.limit, startBefore = _ref37.startBefore;
              return _context19.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                reverse_collection_bids_sorted_by_price: {
                  collection: collection,
                  limit: limit,
                  start_before: startBefore
                }
              }));
            case 2:
            case "end":
              return _context19.stop();
          }
        }
      }, _callee19);
    }));
    return function (_x19) {
      return _ref38.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "askHooks", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee20() {
    return _regenerator["default"].wrap(function _callee20$(_context20) {
      while (1) {
        switch (_context20.prev = _context20.next) {
          case 0:
            return _context20.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              ask_hooks: {}
            }));
          case 1:
          case "end":
            return _context20.stop();
        }
      }
    }, _callee20);
  })));
  (0, _defineProperty2["default"])(this, "bidHooks", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee21() {
    return _regenerator["default"].wrap(function _callee21$(_context21) {
      while (1) {
        switch (_context21.prev = _context21.next) {
          case 0:
            return _context21.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              bid_hooks: {}
            }));
          case 1:
          case "end":
            return _context21.stop();
        }
      }
    }, _callee21);
  })));
  (0, _defineProperty2["default"])(this, "saleHooks", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee22() {
    return _regenerator["default"].wrap(function _callee22$(_context22) {
      while (1) {
        switch (_context22.prev = _context22.next) {
          case 0:
            return _context22.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              sale_hooks: {}
            }));
          case 1:
          case "end":
            return _context22.stop();
        }
      }
    }, _callee22);
  })));
  (0, _defineProperty2["default"])(this, "params", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee23() {
    return _regenerator["default"].wrap(function _callee23$(_context23) {
      while (1) {
        switch (_context23.prev = _context23.next) {
          case 0:
            return _context23.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              params: {}
            }));
          case 1:
          case "end":
            return _context23.stop();
        }
      }
    }, _callee23);
  })));
  this.client = client;
  this.contractAddress = contractAddress;
  this.collections = this.collections.bind(this);
  this.ask = this.ask.bind(this);
  this.asks = this.asks.bind(this);
  this.reverseAsks = this.reverseAsks.bind(this);
  this.asksSortedByPrice = this.asksSortedByPrice.bind(this);
  this.reverseAsksSortedByPrice = this.reverseAsksSortedByPrice.bind(this);
  this.askCount = this.askCount.bind(this);
  this.asksBySeller = this.asksBySeller.bind(this);
  this.bid = this.bid.bind(this);
  this.bidsByBidder = this.bidsByBidder.bind(this);
  this.bidsByBidderSortedByExpiration = this.bidsByBidderSortedByExpiration.bind(this);
  this.bids = this.bids.bind(this);
  this.bidsSortedByPrice = this.bidsSortedByPrice.bind(this);
  this.reverseBidsSortedByPrice = this.reverseBidsSortedByPrice.bind(this);
  this.collectionBid = this.collectionBid.bind(this);
  this.collectionBidsByBidder = this.collectionBidsByBidder.bind(this);
  this.collectionBidsByBidderSortedByExpiration = this.collectionBidsByBidderSortedByExpiration.bind(this);
  this.collectionBidsSortedByPrice = this.collectionBidsSortedByPrice.bind(this);
  this.reverseCollectionBidsSortedByPrice = this.reverseCollectionBidsSortedByPrice.bind(this);
  this.askHooks = this.askHooks.bind(this);
  this.bidHooks = this.bidHooks.bind(this);
  this.saleHooks = this.saleHooks.bind(this);
  this.params = this.params.bind(this);
});
exports.MarketplaceQueryClient = MarketplaceQueryClient;
var MarketplaceClient = /*#__PURE__*/function (_MarketplaceQueryClie) {
  (0, _inherits2["default"])(MarketplaceClient, _MarketplaceQueryClie);
  var _super = _createSuper(MarketplaceClient);
  function MarketplaceClient(client, sender, contractAddress) {
    var _this2;
    (0, _classCallCheck2["default"])(this, MarketplaceClient);
    _this2 = _super.call(this, client, contractAddress);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "client", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sender", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "contractAddress", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "setAsk", /*#__PURE__*/function () {
      var _ref44 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee24(_ref43) {
        var collection,
          expires,
          findersFeeBps,
          fundsRecipient,
          price,
          reserveFor,
          saleType,
          tokenId,
          fee,
          memo,
          _funds,
          _args24 = arguments;
        return _regenerator["default"].wrap(function _callee24$(_context24) {
          while (1) {
            switch (_context24.prev = _context24.next) {
              case 0:
                collection = _ref43.collection, expires = _ref43.expires, findersFeeBps = _ref43.findersFeeBps, fundsRecipient = _ref43.fundsRecipient, price = _ref43.price, reserveFor = _ref43.reserveFor, saleType = _ref43.saleType, tokenId = _ref43.tokenId;
                fee = _args24.length > 1 && _args24[1] !== undefined ? _args24[1] : "auto";
                memo = _args24.length > 2 ? _args24[2] : undefined;
                _funds = _args24.length > 3 ? _args24[3] : undefined;
                _context24.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  set_ask: {
                    collection: collection,
                    expires: expires,
                    finders_fee_bps: findersFeeBps,
                    funds_recipient: fundsRecipient,
                    price: price,
                    reserve_for: reserveFor,
                    sale_type: saleType,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context24.abrupt("return", _context24.sent);
              case 7:
              case "end":
                return _context24.stop();
            }
          }
        }, _callee24);
      }));
      return function (_x20) {
        return _ref44.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeAsk", /*#__PURE__*/function () {
      var _ref46 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee25(_ref45) {
        var collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args25 = arguments;
        return _regenerator["default"].wrap(function _callee25$(_context25) {
          while (1) {
            switch (_context25.prev = _context25.next) {
              case 0:
                collection = _ref45.collection, tokenId = _ref45.tokenId;
                fee = _args25.length > 1 && _args25[1] !== undefined ? _args25[1] : "auto";
                memo = _args25.length > 2 ? _args25[2] : undefined;
                _funds = _args25.length > 3 ? _args25[3] : undefined;
                _context25.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  remove_ask: {
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context25.abrupt("return", _context25.sent);
              case 7:
              case "end":
                return _context25.stop();
            }
          }
        }, _callee25);
      }));
      return function (_x21) {
        return _ref46.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateAskPrice", /*#__PURE__*/function () {
      var _ref48 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee26(_ref47) {
        var collection,
          price,
          tokenId,
          fee,
          memo,
          _funds,
          _args26 = arguments;
        return _regenerator["default"].wrap(function _callee26$(_context26) {
          while (1) {
            switch (_context26.prev = _context26.next) {
              case 0:
                collection = _ref47.collection, price = _ref47.price, tokenId = _ref47.tokenId;
                fee = _args26.length > 1 && _args26[1] !== undefined ? _args26[1] : "auto";
                memo = _args26.length > 2 ? _args26[2] : undefined;
                _funds = _args26.length > 3 ? _args26[3] : undefined;
                _context26.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_ask_price: {
                    collection: collection,
                    price: price,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context26.abrupt("return", _context26.sent);
              case 7:
              case "end":
                return _context26.stop();
            }
          }
        }, _callee26);
      }));
      return function (_x22) {
        return _ref48.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "setBid", /*#__PURE__*/function () {
      var _ref50 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee27(_ref49) {
        var collection,
          expires,
          finder,
          findersFeeBps,
          saleType,
          tokenId,
          fee,
          memo,
          _funds,
          _args27 = arguments;
        return _regenerator["default"].wrap(function _callee27$(_context27) {
          while (1) {
            switch (_context27.prev = _context27.next) {
              case 0:
                collection = _ref49.collection, expires = _ref49.expires, finder = _ref49.finder, findersFeeBps = _ref49.findersFeeBps, saleType = _ref49.saleType, tokenId = _ref49.tokenId;
                fee = _args27.length > 1 && _args27[1] !== undefined ? _args27[1] : "auto";
                memo = _args27.length > 2 ? _args27[2] : undefined;
                _funds = _args27.length > 3 ? _args27[3] : undefined;
                _context27.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  set_bid: {
                    collection: collection,
                    expires: expires,
                    finder: finder,
                    finders_fee_bps: findersFeeBps,
                    sale_type: saleType,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context27.abrupt("return", _context27.sent);
              case 7:
              case "end":
                return _context27.stop();
            }
          }
        }, _callee27);
      }));
      return function (_x23) {
        return _ref50.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "buyNow", /*#__PURE__*/function () {
      var _ref52 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee28(_ref51) {
        var collection,
          expires,
          finder,
          findersFeeBps,
          tokenId,
          fee,
          memo,
          _funds,
          _args28 = arguments;
        return _regenerator["default"].wrap(function _callee28$(_context28) {
          while (1) {
            switch (_context28.prev = _context28.next) {
              case 0:
                collection = _ref51.collection, expires = _ref51.expires, finder = _ref51.finder, findersFeeBps = _ref51.findersFeeBps, tokenId = _ref51.tokenId;
                fee = _args28.length > 1 && _args28[1] !== undefined ? _args28[1] : "auto";
                memo = _args28.length > 2 ? _args28[2] : undefined;
                _funds = _args28.length > 3 ? _args28[3] : undefined;
                _context28.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  buy_now: {
                    collection: collection,
                    expires: expires,
                    finder: finder,
                    finders_fee_bps: findersFeeBps,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context28.abrupt("return", _context28.sent);
              case 7:
              case "end":
                return _context28.stop();
            }
          }
        }, _callee28);
      }));
      return function (_x24) {
        return _ref52.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeBid", /*#__PURE__*/function () {
      var _ref54 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee29(_ref53) {
        var collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args29 = arguments;
        return _regenerator["default"].wrap(function _callee29$(_context29) {
          while (1) {
            switch (_context29.prev = _context29.next) {
              case 0:
                collection = _ref53.collection, tokenId = _ref53.tokenId;
                fee = _args29.length > 1 && _args29[1] !== undefined ? _args29[1] : "auto";
                memo = _args29.length > 2 ? _args29[2] : undefined;
                _funds = _args29.length > 3 ? _args29[3] : undefined;
                _context29.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  remove_bid: {
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context29.abrupt("return", _context29.sent);
              case 7:
              case "end":
                return _context29.stop();
            }
          }
        }, _callee29);
      }));
      return function (_x25) {
        return _ref54.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "acceptBid", /*#__PURE__*/function () {
      var _ref56 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee30(_ref55) {
        var bidder,
          collection,
          finder,
          tokenId,
          fee,
          memo,
          _funds,
          _args30 = arguments;
        return _regenerator["default"].wrap(function _callee30$(_context30) {
          while (1) {
            switch (_context30.prev = _context30.next) {
              case 0:
                bidder = _ref55.bidder, collection = _ref55.collection, finder = _ref55.finder, tokenId = _ref55.tokenId;
                fee = _args30.length > 1 && _args30[1] !== undefined ? _args30[1] : "auto";
                memo = _args30.length > 2 ? _args30[2] : undefined;
                _funds = _args30.length > 3 ? _args30[3] : undefined;
                _context30.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  accept_bid: {
                    bidder: bidder,
                    collection: collection,
                    finder: finder,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context30.abrupt("return", _context30.sent);
              case 7:
              case "end":
                return _context30.stop();
            }
          }
        }, _callee30);
      }));
      return function (_x26) {
        return _ref56.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "rejectBid", /*#__PURE__*/function () {
      var _ref58 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee31(_ref57) {
        var bidder,
          collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args31 = arguments;
        return _regenerator["default"].wrap(function _callee31$(_context31) {
          while (1) {
            switch (_context31.prev = _context31.next) {
              case 0:
                bidder = _ref57.bidder, collection = _ref57.collection, tokenId = _ref57.tokenId;
                fee = _args31.length > 1 && _args31[1] !== undefined ? _args31[1] : "auto";
                memo = _args31.length > 2 ? _args31[2] : undefined;
                _funds = _args31.length > 3 ? _args31[3] : undefined;
                _context31.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  reject_bid: {
                    bidder: bidder,
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context31.abrupt("return", _context31.sent);
              case 7:
              case "end":
                return _context31.stop();
            }
          }
        }, _callee31);
      }));
      return function (_x27) {
        return _ref58.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "setCollectionBid", /*#__PURE__*/function () {
      var _ref60 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee32(_ref59) {
        var collection,
          expires,
          findersFeeBps,
          fee,
          memo,
          _funds,
          _args32 = arguments;
        return _regenerator["default"].wrap(function _callee32$(_context32) {
          while (1) {
            switch (_context32.prev = _context32.next) {
              case 0:
                collection = _ref59.collection, expires = _ref59.expires, findersFeeBps = _ref59.findersFeeBps;
                fee = _args32.length > 1 && _args32[1] !== undefined ? _args32[1] : "auto";
                memo = _args32.length > 2 ? _args32[2] : undefined;
                _funds = _args32.length > 3 ? _args32[3] : undefined;
                _context32.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  set_collection_bid: {
                    collection: collection,
                    expires: expires,
                    finders_fee_bps: findersFeeBps
                  }
                }, fee, memo, _funds);
              case 6:
                return _context32.abrupt("return", _context32.sent);
              case 7:
              case "end":
                return _context32.stop();
            }
          }
        }, _callee32);
      }));
      return function (_x28) {
        return _ref60.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeCollectionBid", /*#__PURE__*/function () {
      var _ref62 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee33(_ref61) {
        var collection,
          fee,
          memo,
          _funds,
          _args33 = arguments;
        return _regenerator["default"].wrap(function _callee33$(_context33) {
          while (1) {
            switch (_context33.prev = _context33.next) {
              case 0:
                collection = _ref61.collection;
                fee = _args33.length > 1 && _args33[1] !== undefined ? _args33[1] : "auto";
                memo = _args33.length > 2 ? _args33[2] : undefined;
                _funds = _args33.length > 3 ? _args33[3] : undefined;
                _context33.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  remove_collection_bid: {
                    collection: collection
                  }
                }, fee, memo, _funds);
              case 6:
                return _context33.abrupt("return", _context33.sent);
              case 7:
              case "end":
                return _context33.stop();
            }
          }
        }, _callee33);
      }));
      return function (_x29) {
        return _ref62.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "acceptCollectionBid", /*#__PURE__*/function () {
      var _ref64 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee34(_ref63) {
        var bidder,
          collection,
          finder,
          tokenId,
          fee,
          memo,
          _funds,
          _args34 = arguments;
        return _regenerator["default"].wrap(function _callee34$(_context34) {
          while (1) {
            switch (_context34.prev = _context34.next) {
              case 0:
                bidder = _ref63.bidder, collection = _ref63.collection, finder = _ref63.finder, tokenId = _ref63.tokenId;
                fee = _args34.length > 1 && _args34[1] !== undefined ? _args34[1] : "auto";
                memo = _args34.length > 2 ? _args34[2] : undefined;
                _funds = _args34.length > 3 ? _args34[3] : undefined;
                _context34.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  accept_collection_bid: {
                    bidder: bidder,
                    collection: collection,
                    finder: finder,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context34.abrupt("return", _context34.sent);
              case 7:
              case "end":
                return _context34.stop();
            }
          }
        }, _callee34);
      }));
      return function (_x30) {
        return _ref64.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "syncAsk", /*#__PURE__*/function () {
      var _ref66 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee35(_ref65) {
        var collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args35 = arguments;
        return _regenerator["default"].wrap(function _callee35$(_context35) {
          while (1) {
            switch (_context35.prev = _context35.next) {
              case 0:
                collection = _ref65.collection, tokenId = _ref65.tokenId;
                fee = _args35.length > 1 && _args35[1] !== undefined ? _args35[1] : "auto";
                memo = _args35.length > 2 ? _args35[2] : undefined;
                _funds = _args35.length > 3 ? _args35[3] : undefined;
                _context35.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  sync_ask: {
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context35.abrupt("return", _context35.sent);
              case 7:
              case "end":
                return _context35.stop();
            }
          }
        }, _callee35);
      }));
      return function (_x31) {
        return _ref66.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeStaleAsk", /*#__PURE__*/function () {
      var _ref68 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee36(_ref67) {
        var collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args36 = arguments;
        return _regenerator["default"].wrap(function _callee36$(_context36) {
          while (1) {
            switch (_context36.prev = _context36.next) {
              case 0:
                collection = _ref67.collection, tokenId = _ref67.tokenId;
                fee = _args36.length > 1 && _args36[1] !== undefined ? _args36[1] : "auto";
                memo = _args36.length > 2 ? _args36[2] : undefined;
                _funds = _args36.length > 3 ? _args36[3] : undefined;
                _context36.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  remove_stale_ask: {
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context36.abrupt("return", _context36.sent);
              case 7:
              case "end":
                return _context36.stop();
            }
          }
        }, _callee36);
      }));
      return function (_x32) {
        return _ref68.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeStaleBid", /*#__PURE__*/function () {
      var _ref70 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee37(_ref69) {
        var bidder,
          collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args37 = arguments;
        return _regenerator["default"].wrap(function _callee37$(_context37) {
          while (1) {
            switch (_context37.prev = _context37.next) {
              case 0:
                bidder = _ref69.bidder, collection = _ref69.collection, tokenId = _ref69.tokenId;
                fee = _args37.length > 1 && _args37[1] !== undefined ? _args37[1] : "auto";
                memo = _args37.length > 2 ? _args37[2] : undefined;
                _funds = _args37.length > 3 ? _args37[3] : undefined;
                _context37.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  remove_stale_bid: {
                    bidder: bidder,
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context37.abrupt("return", _context37.sent);
              case 7:
              case "end":
                return _context37.stop();
            }
          }
        }, _callee37);
      }));
      return function (_x33) {
        return _ref70.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeStaleCollectionBid", /*#__PURE__*/function () {
      var _ref72 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee38(_ref71) {
        var bidder,
          collection,
          fee,
          memo,
          _funds,
          _args38 = arguments;
        return _regenerator["default"].wrap(function _callee38$(_context38) {
          while (1) {
            switch (_context38.prev = _context38.next) {
              case 0:
                bidder = _ref71.bidder, collection = _ref71.collection;
                fee = _args38.length > 1 && _args38[1] !== undefined ? _args38[1] : "auto";
                memo = _args38.length > 2 ? _args38[2] : undefined;
                _funds = _args38.length > 3 ? _args38[3] : undefined;
                _context38.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  remove_stale_collection_bid: {
                    bidder: bidder,
                    collection: collection
                  }
                }, fee, memo, _funds);
              case 6:
                return _context38.abrupt("return", _context38.sent);
              case 7:
              case "end":
                return _context38.stop();
            }
          }
        }, _callee38);
      }));
      return function (_x34) {
        return _ref72.apply(this, arguments);
      };
    }());
    _this2.client = client;
    _this2.sender = sender;
    _this2.contractAddress = contractAddress;
    _this2.setAsk = _this2.setAsk.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeAsk = _this2.removeAsk.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateAskPrice = _this2.updateAskPrice.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.setBid = _this2.setBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.buyNow = _this2.buyNow.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeBid = _this2.removeBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.acceptBid = _this2.acceptBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.rejectBid = _this2.rejectBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.setCollectionBid = _this2.setCollectionBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeCollectionBid = _this2.removeCollectionBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.acceptCollectionBid = _this2.acceptCollectionBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.syncAsk = _this2.syncAsk.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeStaleAsk = _this2.removeStaleAsk.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeStaleBid = _this2.removeStaleBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeStaleCollectionBid = _this2.removeStaleCollectionBid.bind((0, _assertThisInitialized2["default"])(_this2));
    return _this2;
  }
  return (0, _createClass2["default"])(MarketplaceClient);
}(MarketplaceQueryClient);
exports.MarketplaceClient = MarketplaceClient;
