// stargazejs/main/codegen/ReserveAuction.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ReserveAuctionQueryClient = exports.ReserveAuctionClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var ReserveAuctionQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function ReserveAuctionQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, ReserveAuctionQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "config", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              config: {}
            }));
          case 1:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  })));
  (0, _defineProperty2["default"])(this, "auction", /*#__PURE__*/function () {
    var _ref3 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(_ref2) {
      var collection, tokenId;
      return _regenerator["default"].wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              collection = _ref2.collection, tokenId = _ref2.tokenId;
              return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                auction: {
                  collection: collection,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }));
    return function (_x) {
      return _ref3.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "auctionsBySeller", /*#__PURE__*/function () {
    var _ref5 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(_ref4) {
      var queryOptions, seller;
      return _regenerator["default"].wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              queryOptions = _ref4.queryOptions, seller = _ref4.seller;
              return _context3.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                auctions_by_seller: {
                  query_options: queryOptions,
                  seller: seller
                }
              }));
            case 2:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3);
    }));
    return function (_x2) {
      return _ref5.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "auctionsByEndTime", /*#__PURE__*/function () {
    var _ref7 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(_ref6) {
      var endTime, queryOptions;
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              endTime = _ref6.endTime, queryOptions = _ref6.queryOptions;
              return _context4.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                auctions_by_end_time: {
                  end_time: endTime,
                  query_options: queryOptions
                }
              }));
            case 2:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4);
    }));
    return function (_x3) {
      return _ref7.apply(this, arguments);
    };
  }());
  this.client = client;
  this.contractAddress = contractAddress;
  this.config = this.config.bind(this);
  this.auction = this.auction.bind(this);
  this.auctionsBySeller = this.auctionsBySeller.bind(this);
  this.auctionsByEndTime = this.auctionsByEndTime.bind(this);
});
exports.ReserveAuctionQueryClient = ReserveAuctionQueryClient;
var ReserveAuctionClient = /*#__PURE__*/function (_ReserveAuctionQueryC) {
  (0, _inherits2["default"])(ReserveAuctionClient, _ReserveAuctionQueryC);
  var _super = _createSuper(ReserveAuctionClient);
  function ReserveAuctionClient(client, sender, contractAddress) {
    var _this2;
    (0, _classCallCheck2["default"])(this, ReserveAuctionClient);
    _this2 = _super.call(this, client, contractAddress);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "client", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sender", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "contractAddress", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "createAuction", /*#__PURE__*/function () {
      var _ref9 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(_ref8) {
        var collection,
          endTime,
          reservePrice,
          sellerFundsRecipient,
          startTime,
          tokenId,
          fee,
          memo,
          _funds,
          _args5 = arguments;
        return _regenerator["default"].wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                collection = _ref8.collection, endTime = _ref8.endTime, reservePrice = _ref8.reservePrice, sellerFundsRecipient = _ref8.sellerFundsRecipient, startTime = _ref8.startTime, tokenId = _ref8.tokenId;
                fee = _args5.length > 1 && _args5[1] !== undefined ? _args5[1] : "auto";
                memo = _args5.length > 2 ? _args5[2] : undefined;
                _funds = _args5.length > 3 ? _args5[3] : undefined;
                _context5.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  create_auction: {
                    collection: collection,
                    end_time: endTime,
                    reserve_price: reservePrice,
                    seller_funds_recipient: sellerFundsRecipient,
                    start_time: startTime,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context5.abrupt("return", _context5.sent);
              case 7:
              case "end":
                return _context5.stop();
            }
          }
        }, _callee5);
      }));
      return function (_x4) {
        return _ref9.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateReservePrice", /*#__PURE__*/function () {
      var _ref11 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6(_ref10) {
        var collection,
          reservePrice,
          tokenId,
          fee,
          memo,
          _funds,
          _args6 = arguments;
        return _regenerator["default"].wrap(function _callee6$(_context6) {
          while (1) {
            switch (_context6.prev = _context6.next) {
              case 0:
                collection = _ref10.collection, reservePrice = _ref10.reservePrice, tokenId = _ref10.tokenId;
                fee = _args6.length > 1 && _args6[1] !== undefined ? _args6[1] : "auto";
                memo = _args6.length > 2 ? _args6[2] : undefined;
                _funds = _args6.length > 3 ? _args6[3] : undefined;
                _context6.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_reserve_price: {
                    collection: collection,
                    reserve_price: reservePrice,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context6.abrupt("return", _context6.sent);
              case 7:
              case "end":
                return _context6.stop();
            }
          }
        }, _callee6);
      }));
      return function (_x5) {
        return _ref11.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "cancelAuction", /*#__PURE__*/function () {
      var _ref13 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7(_ref12) {
        var collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args7 = arguments;
        return _regenerator["default"].wrap(function _callee7$(_context7) {
          while (1) {
            switch (_context7.prev = _context7.next) {
              case 0:
                collection = _ref12.collection, tokenId = _ref12.tokenId;
                fee = _args7.length > 1 && _args7[1] !== undefined ? _args7[1] : "auto";
                memo = _args7.length > 2 ? _args7[2] : undefined;
                _funds = _args7.length > 3 ? _args7[3] : undefined;
                _context7.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  cancel_auction: {
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context7.abrupt("return", _context7.sent);
              case 7:
              case "end":
                return _context7.stop();
            }
          }
        }, _callee7);
      }));
      return function (_x6) {
        return _ref13.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "placeBid", /*#__PURE__*/function () {
      var _ref15 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee8(_ref14) {
        var collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args8 = arguments;
        return _regenerator["default"].wrap(function _callee8$(_context8) {
          while (1) {
            switch (_context8.prev = _context8.next) {
              case 0:
                collection = _ref14.collection, tokenId = _ref14.tokenId;
                fee = _args8.length > 1 && _args8[1] !== undefined ? _args8[1] : "auto";
                memo = _args8.length > 2 ? _args8[2] : undefined;
                _funds = _args8.length > 3 ? _args8[3] : undefined;
                _context8.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  place_bid: {
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context8.abrupt("return", _context8.sent);
              case 7:
              case "end":
                return _context8.stop();
            }
          }
        }, _callee8);
      }));
      return function (_x7) {
        return _ref15.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "settleAuction", /*#__PURE__*/function () {
      var _ref17 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee9(_ref16) {
        var collection,
          tokenId,
          fee,
          memo,
          _funds,
          _args9 = arguments;
        return _regenerator["default"].wrap(function _callee9$(_context9) {
          while (1) {
            switch (_context9.prev = _context9.next) {
              case 0:
                collection = _ref16.collection, tokenId = _ref16.tokenId;
                fee = _args9.length > 1 && _args9[1] !== undefined ? _args9[1] : "auto";
                memo = _args9.length > 2 ? _args9[2] : undefined;
                _funds = _args9.length > 3 ? _args9[3] : undefined;
                _context9.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  settle_auction: {
                    collection: collection,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context9.abrupt("return", _context9.sent);
              case 7:
              case "end":
                return _context9.stop();
            }
          }
        }, _callee9);
      }));
      return function (_x8) {
        return _ref17.apply(this, arguments);
      };
    }());
    _this2.client = client;
    _this2.sender = sender;
    _this2.contractAddress = contractAddress;
    _this2.createAuction = _this2.createAuction.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateReservePrice = _this2.updateReservePrice.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.cancelAuction = _this2.cancelAuction.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.placeBid = _this2.placeBid.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.settleAuction = _this2.settleAuction.bind((0, _assertThisInitialized2["default"])(_this2));
    return _this2;
  }
  return (0, _createClass2["default"])(ReserveAuctionClient);
}(ReserveAuctionQueryClient);
exports.ReserveAuctionClient = ReserveAuctionClient;
