// stargazejs/main/codegen/Sg721Updatable.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sg721UpdatableQueryClient = exports.Sg721UpdatableClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var Sg721UpdatableQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function Sg721UpdatableQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, Sg721UpdatableQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "ownerOf", /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(_ref) {
      var includeExpired, tokenId;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              includeExpired = _ref.includeExpired, tokenId = _ref.tokenId;
              return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                owner_of: {
                  include_expired: includeExpired,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "approval", /*#__PURE__*/function () {
    var _ref4 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(_ref3) {
      var includeExpired, spender, tokenId;
      return _regenerator["default"].wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              includeExpired = _ref3.includeExpired, spender = _ref3.spender, tokenId = _ref3.tokenId;
              return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                approval: {
                  include_expired: includeExpired,
                  spender: spender,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }));
    return function (_x2) {
      return _ref4.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "approvals", /*#__PURE__*/function () {
    var _ref6 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(_ref5) {
      var includeExpired, tokenId;
      return _regenerator["default"].wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              includeExpired = _ref5.includeExpired, tokenId = _ref5.tokenId;
              return _context3.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                approvals: {
                  include_expired: includeExpired,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3);
    }));
    return function (_x3) {
      return _ref6.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "allOperators", /*#__PURE__*/function () {
    var _ref8 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(_ref7) {
      var includeExpired, limit, owner, startAfter;
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              includeExpired = _ref7.includeExpired, limit = _ref7.limit, owner = _ref7.owner, startAfter = _ref7.startAfter;
              return _context4.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                all_operators: {
                  include_expired: includeExpired,
                  limit: limit,
                  owner: owner,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4);
    }));
    return function (_x4) {
      return _ref8.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "numTokens", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5() {
    return _regenerator["default"].wrap(function _callee5$(_context5) {
      while (1) {
        switch (_context5.prev = _context5.next) {
          case 0:
            return _context5.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              num_tokens: {}
            }));
          case 1:
          case "end":
            return _context5.stop();
        }
      }
    }, _callee5);
  })));
  (0, _defineProperty2["default"])(this, "contractInfo", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6() {
    return _regenerator["default"].wrap(function _callee6$(_context6) {
      while (1) {
        switch (_context6.prev = _context6.next) {
          case 0:
            return _context6.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              contract_info: {}
            }));
          case 1:
          case "end":
            return _context6.stop();
        }
      }
    }, _callee6);
  })));
  (0, _defineProperty2["default"])(this, "nftInfo", /*#__PURE__*/function () {
    var _ref12 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7(_ref11) {
      var tokenId;
      return _regenerator["default"].wrap(function _callee7$(_context7) {
        while (1) {
          switch (_context7.prev = _context7.next) {
            case 0:
              tokenId = _ref11.tokenId;
              return _context7.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                nft_info: {
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context7.stop();
          }
        }
      }, _callee7);
    }));
    return function (_x5) {
      return _ref12.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "allNftInfo", /*#__PURE__*/function () {
    var _ref14 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee8(_ref13) {
      var includeExpired, tokenId;
      return _regenerator["default"].wrap(function _callee8$(_context8) {
        while (1) {
          switch (_context8.prev = _context8.next) {
            case 0:
              includeExpired = _ref13.includeExpired, tokenId = _ref13.tokenId;
              return _context8.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                all_nft_info: {
                  include_expired: includeExpired,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context8.stop();
          }
        }
      }, _callee8);
    }));
    return function (_x6) {
      return _ref14.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "tokens", /*#__PURE__*/function () {
    var _ref16 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee9(_ref15) {
      var limit, owner, startAfter;
      return _regenerator["default"].wrap(function _callee9$(_context9) {
        while (1) {
          switch (_context9.prev = _context9.next) {
            case 0:
              limit = _ref15.limit, owner = _ref15.owner, startAfter = _ref15.startAfter;
              return _context9.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                tokens: {
                  limit: limit,
                  owner: owner,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context9.stop();
          }
        }
      }, _callee9);
    }));
    return function (_x7) {
      return _ref16.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "allTokens", /*#__PURE__*/function () {
    var _ref18 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee10(_ref17) {
      var limit, startAfter;
      return _regenerator["default"].wrap(function _callee10$(_context10) {
        while (1) {
          switch (_context10.prev = _context10.next) {
            case 0:
              limit = _ref17.limit, startAfter = _ref17.startAfter;
              return _context10.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                all_tokens: {
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context10.stop();
          }
        }
      }, _callee10);
    }));
    return function (_x8) {
      return _ref18.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "minter", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee11() {
    return _regenerator["default"].wrap(function _callee11$(_context11) {
      while (1) {
        switch (_context11.prev = _context11.next) {
          case 0:
            return _context11.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              minter: {}
            }));
          case 1:
          case "end":
            return _context11.stop();
        }
      }
    }, _callee11);
  })));
  (0, _defineProperty2["default"])(this, "collectionInfo", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee12() {
    return _regenerator["default"].wrap(function _callee12$(_context12) {
      while (1) {
        switch (_context12.prev = _context12.next) {
          case 0:
            return _context12.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              collection_info: {}
            }));
          case 1:
          case "end":
            return _context12.stop();
        }
      }
    }, _callee12);
  })));
  this.client = client;
  this.contractAddress = contractAddress;
  this.ownerOf = this.ownerOf.bind(this);
  this.approval = this.approval.bind(this);
  this.approvals = this.approvals.bind(this);
  this.allOperators = this.allOperators.bind(this);
  this.numTokens = this.numTokens.bind(this);
  this.contractInfo = this.contractInfo.bind(this);
  this.nftInfo = this.nftInfo.bind(this);
  this.allNftInfo = this.allNftInfo.bind(this);
  this.tokens = this.tokens.bind(this);
  this.allTokens = this.allTokens.bind(this);
  this.minter = this.minter.bind(this);
  this.collectionInfo = this.collectionInfo.bind(this);
});
exports.Sg721UpdatableQueryClient = Sg721UpdatableQueryClient;
var Sg721UpdatableClient = /*#__PURE__*/function (_Sg721UpdatableQueryC) {
  (0, _inherits2["default"])(Sg721UpdatableClient, _Sg721UpdatableQueryC);
  var _super = _createSuper(Sg721UpdatableClient);
  function Sg721UpdatableClient(client, sender, contractAddress) {
    var _this2;
    (0, _classCallCheck2["default"])(this, Sg721UpdatableClient);
    _this2 = _super.call(this, client, contractAddress);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "client", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sender", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "contractAddress", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "freezeTokenMetadata", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee13() {
      var fee,
        memo,
        _funds,
        _args13 = arguments;
      return _regenerator["default"].wrap(function _callee13$(_context13) {
        while (1) {
          switch (_context13.prev = _context13.next) {
            case 0:
              fee = _args13.length > 0 && _args13[0] !== undefined ? _args13[0] : "auto";
              memo = _args13.length > 1 ? _args13[1] : undefined;
              _funds = _args13.length > 2 ? _args13[2] : undefined;
              _context13.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                freeze_token_metadata: {}
              }, fee, memo, _funds);
            case 5:
              return _context13.abrupt("return", _context13.sent);
            case 6:
            case "end":
              return _context13.stop();
          }
        }
      }, _callee13);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateTokenMetadata", /*#__PURE__*/function () {
      var _ref23 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee14(_ref22) {
        var tokenId,
          tokenUri,
          fee,
          memo,
          _funds,
          _args14 = arguments;
        return _regenerator["default"].wrap(function _callee14$(_context14) {
          while (1) {
            switch (_context14.prev = _context14.next) {
              case 0:
                tokenId = _ref22.tokenId, tokenUri = _ref22.tokenUri;
                fee = _args14.length > 1 && _args14[1] !== undefined ? _args14[1] : "auto";
                memo = _args14.length > 2 ? _args14[2] : undefined;
                _funds = _args14.length > 3 ? _args14[3] : undefined;
                _context14.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_token_metadata: {
                    token_id: tokenId,
                    token_uri: tokenUri
                  }
                }, fee, memo, _funds);
              case 6:
                return _context14.abrupt("return", _context14.sent);
              case 7:
              case "end":
                return _context14.stop();
            }
          }
        }, _callee14);
      }));
      return function (_x9) {
        return _ref23.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "enableUpdatable", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee15() {
      var fee,
        memo,
        _funds,
        _args15 = arguments;
      return _regenerator["default"].wrap(function _callee15$(_context15) {
        while (1) {
          switch (_context15.prev = _context15.next) {
            case 0:
              fee = _args15.length > 0 && _args15[0] !== undefined ? _args15[0] : "auto";
              memo = _args15.length > 1 ? _args15[1] : undefined;
              _funds = _args15.length > 2 ? _args15[2] : undefined;
              _context15.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                enable_updatable: {}
              }, fee, memo, _funds);
            case 5:
              return _context15.abrupt("return", _context15.sent);
            case 6:
            case "end":
              return _context15.stop();
          }
        }
      }, _callee15);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "transferNft", /*#__PURE__*/function () {
      var _ref26 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee16(_ref25) {
        var recipient,
          tokenId,
          fee,
          memo,
          _funds,
          _args16 = arguments;
        return _regenerator["default"].wrap(function _callee16$(_context16) {
          while (1) {
            switch (_context16.prev = _context16.next) {
              case 0:
                recipient = _ref25.recipient, tokenId = _ref25.tokenId;
                fee = _args16.length > 1 && _args16[1] !== undefined ? _args16[1] : "auto";
                memo = _args16.length > 2 ? _args16[2] : undefined;
                _funds = _args16.length > 3 ? _args16[3] : undefined;
                _context16.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  transfer_nft: {
                    recipient: recipient,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context16.abrupt("return", _context16.sent);
              case 7:
              case "end":
                return _context16.stop();
            }
          }
        }, _callee16);
      }));
      return function (_x10) {
        return _ref26.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sendNft", /*#__PURE__*/function () {
      var _ref28 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee17(_ref27) {
        var contract,
          msg,
          tokenId,
          fee,
          memo,
          _funds,
          _args17 = arguments;
        return _regenerator["default"].wrap(function _callee17$(_context17) {
          while (1) {
            switch (_context17.prev = _context17.next) {
              case 0:
                contract = _ref27.contract, msg = _ref27.msg, tokenId = _ref27.tokenId;
                fee = _args17.length > 1 && _args17[1] !== undefined ? _args17[1] : "auto";
                memo = _args17.length > 2 ? _args17[2] : undefined;
                _funds = _args17.length > 3 ? _args17[3] : undefined;
                _context17.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  send_nft: {
                    contract: contract,
                    msg: msg,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context17.abrupt("return", _context17.sent);
              case 7:
              case "end":
                return _context17.stop();
            }
          }
        }, _callee17);
      }));
      return function (_x11) {
        return _ref28.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "approve", /*#__PURE__*/function () {
      var _ref30 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee18(_ref29) {
        var expires,
          spender,
          tokenId,
          fee,
          memo,
          _funds,
          _args18 = arguments;
        return _regenerator["default"].wrap(function _callee18$(_context18) {
          while (1) {
            switch (_context18.prev = _context18.next) {
              case 0:
                expires = _ref29.expires, spender = _ref29.spender, tokenId = _ref29.tokenId;
                fee = _args18.length > 1 && _args18[1] !== undefined ? _args18[1] : "auto";
                memo = _args18.length > 2 ? _args18[2] : undefined;
                _funds = _args18.length > 3 ? _args18[3] : undefined;
                _context18.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  approve: {
                    expires: expires,
                    spender: spender,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context18.abrupt("return", _context18.sent);
              case 7:
              case "end":
                return _context18.stop();
            }
          }
        }, _callee18);
      }));
      return function (_x12) {
        return _ref30.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "revoke", /*#__PURE__*/function () {
      var _ref32 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee19(_ref31) {
        var spender,
          tokenId,
          fee,
          memo,
          _funds,
          _args19 = arguments;
        return _regenerator["default"].wrap(function _callee19$(_context19) {
          while (1) {
            switch (_context19.prev = _context19.next) {
              case 0:
                spender = _ref31.spender, tokenId = _ref31.tokenId;
                fee = _args19.length > 1 && _args19[1] !== undefined ? _args19[1] : "auto";
                memo = _args19.length > 2 ? _args19[2] : undefined;
                _funds = _args19.length > 3 ? _args19[3] : undefined;
                _context19.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  revoke: {
                    spender: spender,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context19.abrupt("return", _context19.sent);
              case 7:
              case "end":
                return _context19.stop();
            }
          }
        }, _callee19);
      }));
      return function (_x13) {
        return _ref32.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "approveAll", /*#__PURE__*/function () {
      var _ref34 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee20(_ref33) {
        var expires,
          operator,
          fee,
          memo,
          _funds,
          _args20 = arguments;
        return _regenerator["default"].wrap(function _callee20$(_context20) {
          while (1) {
            switch (_context20.prev = _context20.next) {
              case 0:
                expires = _ref33.expires, operator = _ref33.operator;
                fee = _args20.length > 1 && _args20[1] !== undefined ? _args20[1] : "auto";
                memo = _args20.length > 2 ? _args20[2] : undefined;
                _funds = _args20.length > 3 ? _args20[3] : undefined;
                _context20.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  approve_all: {
                    expires: expires,
                    operator: operator
                  }
                }, fee, memo, _funds);
              case 6:
                return _context20.abrupt("return", _context20.sent);
              case 7:
              case "end":
                return _context20.stop();
            }
          }
        }, _callee20);
      }));
      return function (_x14) {
        return _ref34.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "revokeAll", /*#__PURE__*/function () {
      var _ref36 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee21(_ref35) {
        var operator,
          fee,
          memo,
          _funds,
          _args21 = arguments;
        return _regenerator["default"].wrap(function _callee21$(_context21) {
          while (1) {
            switch (_context21.prev = _context21.next) {
              case 0:
                operator = _ref35.operator;
                fee = _args21.length > 1 && _args21[1] !== undefined ? _args21[1] : "auto";
                memo = _args21.length > 2 ? _args21[2] : undefined;
                _funds = _args21.length > 3 ? _args21[3] : undefined;
                _context21.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  revoke_all: {
                    operator: operator
                  }
                }, fee, memo, _funds);
              case 6:
                return _context21.abrupt("return", _context21.sent);
              case 7:
              case "end":
                return _context21.stop();
            }
          }
        }, _callee21);
      }));
      return function (_x15) {
        return _ref36.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "burn", /*#__PURE__*/function () {
      var _ref38 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee22(_ref37) {
        var tokenId,
          fee,
          memo,
          _funds,
          _args22 = arguments;
        return _regenerator["default"].wrap(function _callee22$(_context22) {
          while (1) {
            switch (_context22.prev = _context22.next) {
              case 0:
                tokenId = _ref37.tokenId;
                fee = _args22.length > 1 && _args22[1] !== undefined ? _args22[1] : "auto";
                memo = _args22.length > 2 ? _args22[2] : undefined;
                _funds = _args22.length > 3 ? _args22[3] : undefined;
                _context22.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  burn: {
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context22.abrupt("return", _context22.sent);
              case 7:
              case "end":
                return _context22.stop();
            }
          }
        }, _callee22);
      }));
      return function (_x16) {
        return _ref38.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateCollectionInfo", /*#__PURE__*/function () {
      var _ref40 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee23(_ref39) {
        var collectionInfo,
          fee,
          memo,
          _funds,
          _args23 = arguments;
        return _regenerator["default"].wrap(function _callee23$(_context23) {
          while (1) {
            switch (_context23.prev = _context23.next) {
              case 0:
                collectionInfo = _ref39.collectionInfo;
                fee = _args23.length > 1 && _args23[1] !== undefined ? _args23[1] : "auto";
                memo = _args23.length > 2 ? _args23[2] : undefined;
                _funds = _args23.length > 3 ? _args23[3] : undefined;
                _context23.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_collection_info: {
                    collection_info: collectionInfo
                  }
                }, fee, memo, _funds);
              case 6:
                return _context23.abrupt("return", _context23.sent);
              case 7:
              case "end":
                return _context23.stop();
            }
          }
        }, _callee23);
      }));
      return function (_x17) {
        return _ref40.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateTradingStartTime", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee24() {
      var fee,
        memo,
        _funds,
        _args24 = arguments;
      return _regenerator["default"].wrap(function _callee24$(_context24) {
        while (1) {
          switch (_context24.prev = _context24.next) {
            case 0:
              fee = _args24.length > 0 && _args24[0] !== undefined ? _args24[0] : "auto";
              memo = _args24.length > 1 ? _args24[1] : undefined;
              _funds = _args24.length > 2 ? _args24[2] : undefined;
              _context24.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                update_trading_start_time: {}
              }, fee, memo, _funds);
            case 5:
              return _context24.abrupt("return", _context24.sent);
            case 6:
            case "end":
              return _context24.stop();
          }
        }
      }, _callee24);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "freezeCollectionInfo", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee25() {
      var fee,
        memo,
        _funds,
        _args25 = arguments;
      return _regenerator["default"].wrap(function _callee25$(_context25) {
        while (1) {
          switch (_context25.prev = _context25.next) {
            case 0:
              fee = _args25.length > 0 && _args25[0] !== undefined ? _args25[0] : "auto";
              memo = _args25.length > 1 ? _args25[1] : undefined;
              _funds = _args25.length > 2 ? _args25[2] : undefined;
              _context25.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                freeze_collection_info: {}
              }, fee, memo, _funds);
            case 5:
              return _context25.abrupt("return", _context25.sent);
            case 6:
            case "end":
              return _context25.stop();
          }
        }
      }, _callee25);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "mint", /*#__PURE__*/function () {
      var _ref44 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee26(_ref43) {
        var extension,
          owner,
          tokenId,
          tokenUri,
          fee,
          memo,
          _funds,
          _args26 = arguments;
        return _regenerator["default"].wrap(function _callee26$(_context26) {
          while (1) {
            switch (_context26.prev = _context26.next) {
              case 0:
                extension = _ref43.extension, owner = _ref43.owner, tokenId = _ref43.tokenId, tokenUri = _ref43.tokenUri;
                fee = _args26.length > 1 && _args26[1] !== undefined ? _args26[1] : "auto";
                memo = _args26.length > 2 ? _args26[2] : undefined;
                _funds = _args26.length > 3 ? _args26[3] : undefined;
                _context26.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  mint: {
                    extension: extension,
                    owner: owner,
                    token_id: tokenId,
                    token_uri: tokenUri
                  }
                }, fee, memo, _funds);
              case 6:
                return _context26.abrupt("return", _context26.sent);
              case 7:
              case "end":
                return _context26.stop();
            }
          }
        }, _callee26);
      }));
      return function (_x18) {
        return _ref44.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "extension", /*#__PURE__*/function () {
      var _ref46 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee27(_ref45) {
        var msg,
          fee,
          memo,
          _funds,
          _args27 = arguments;
        return _regenerator["default"].wrap(function _callee27$(_context27) {
          while (1) {
            switch (_context27.prev = _context27.next) {
              case 0:
                msg = _ref45.msg;
                fee = _args27.length > 1 && _args27[1] !== undefined ? _args27[1] : "auto";
                memo = _args27.length > 2 ? _args27[2] : undefined;
                _funds = _args27.length > 3 ? _args27[3] : undefined;
                _context27.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  extension: {
                    msg: msg
                  }
                }, fee, memo, _funds);
              case 6:
                return _context27.abrupt("return", _context27.sent);
              case 7:
              case "end":
                return _context27.stop();
            }
          }
        }, _callee27);
      }));
      return function (_x19) {
        return _ref46.apply(this, arguments);
      };
    }());
    _this2.client = client;
    _this2.sender = sender;
    _this2.contractAddress = contractAddress;
    _this2.freezeTokenMetadata = _this2.freezeTokenMetadata.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateTokenMetadata = _this2.updateTokenMetadata.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.enableUpdatable = _this2.enableUpdatable.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.transferNft = _this2.transferNft.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.sendNft = _this2.sendNft.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.approve = _this2.approve.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.revoke = _this2.revoke.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.approveAll = _this2.approveAll.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.revokeAll = _this2.revokeAll.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.burn = _this2.burn.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateCollectionInfo = _this2.updateCollectionInfo.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateTradingStartTime = _this2.updateTradingStartTime.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.freezeCollectionInfo = _this2.freezeCollectionInfo.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.mint = _this2.mint.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.extension = _this2.extension.bind((0, _assertThisInitialized2["default"])(_this2));
    return _this2;
  }
  return (0, _createClass2["default"])(Sg721UpdatableClient);
}(Sg721UpdatableQueryClient);
exports.Sg721UpdatableClient = Sg721UpdatableClient;
