// stargazejs/main/codegen/SG721Base.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SG721BaseQueryClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var SG721BaseQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function SG721BaseQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, SG721BaseQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "ownerOf", /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(_ref) {
      var includeExpired, tokenId;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              includeExpired = _ref.includeExpired, tokenId = _ref.tokenId;
              return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                owner_of: {
                  include_expired: includeExpired,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "approval", /*#__PURE__*/function () {
    var _ref4 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(_ref3) {
      var includeExpired, spender, tokenId;
      return _regenerator["default"].wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              includeExpired = _ref3.includeExpired, spender = _ref3.spender, tokenId = _ref3.tokenId;
              return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                approval: {
                  include_expired: includeExpired,
                  spender: spender,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }));
    return function (_x2) {
      return _ref4.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "approvals", /*#__PURE__*/function () {
    var _ref6 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(_ref5) {
      var includeExpired, tokenId;
      return _regenerator["default"].wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              includeExpired = _ref5.includeExpired, tokenId = _ref5.tokenId;
              return _context3.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                approvals: {
                  include_expired: includeExpired,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3);
    }));
    return function (_x3) {
      return _ref6.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "allOperators", /*#__PURE__*/function () {
    var _ref8 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(_ref7) {
      var includeExpired, limit, owner, startAfter;
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              includeExpired = _ref7.includeExpired, limit = _ref7.limit, owner = _ref7.owner, startAfter = _ref7.startAfter;
              return _context4.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                all_operators: {
                  include_expired: includeExpired,
                  limit: limit,
                  owner: owner,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4);
    }));
    return function (_x4) {
      return _ref8.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "numTokens", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5() {
    return _regenerator["default"].wrap(function _callee5$(_context5) {
      while (1) {
        switch (_context5.prev = _context5.next) {
          case 0:
            return _context5.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              num_tokens: {}
            }));
          case 1:
          case "end":
            return _context5.stop();
        }
      }
    }, _callee5);
  })));
  (0, _defineProperty2["default"])(this, "contractInfo", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6() {
    return _regenerator["default"].wrap(function _callee6$(_context6) {
      while (1) {
        switch (_context6.prev = _context6.next) {
          case 0:
            return _context6.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              contract_info: {}
            }));
          case 1:
          case "end":
            return _context6.stop();
        }
      }
    }, _callee6);
  })));
  (0, _defineProperty2["default"])(this, "nftInfo", /*#__PURE__*/function () {
    var _ref12 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7(_ref11) {
      var tokenId;
      return _regenerator["default"].wrap(function _callee7$(_context7) {
        while (1) {
          switch (_context7.prev = _context7.next) {
            case 0:
              tokenId = _ref11.tokenId;
              return _context7.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                nft_info: {
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context7.stop();
          }
        }
      }, _callee7);
    }));
    return function (_x5) {
      return _ref12.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "allNftInfo", /*#__PURE__*/function () {
    var _ref14 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee8(_ref13) {
      var includeExpired, tokenId;
      return _regenerator["default"].wrap(function _callee8$(_context8) {
        while (1) {
          switch (_context8.prev = _context8.next) {
            case 0:
              includeExpired = _ref13.includeExpired, tokenId = _ref13.tokenId;
              return _context8.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                all_nft_info: {
                  include_expired: includeExpired,
                  token_id: tokenId
                }
              }));
            case 2:
            case "end":
              return _context8.stop();
          }
        }
      }, _callee8);
    }));
    return function (_x6) {
      return _ref14.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "tokens", /*#__PURE__*/function () {
    var _ref16 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee9(_ref15) {
      var limit, owner, startAfter;
      return _regenerator["default"].wrap(function _callee9$(_context9) {
        while (1) {
          switch (_context9.prev = _context9.next) {
            case 0:
              limit = _ref15.limit, owner = _ref15.owner, startAfter = _ref15.startAfter;
              return _context9.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                tokens: {
                  limit: limit,
                  owner: owner,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context9.stop();
          }
        }
      }, _callee9);
    }));
    return function (_x7) {
      return _ref16.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "allTokens", /*#__PURE__*/function () {
    var _ref18 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee10(_ref17) {
      var limit, startAfter;
      return _regenerator["default"].wrap(function _callee10$(_context10) {
        while (1) {
          switch (_context10.prev = _context10.next) {
            case 0:
              limit = _ref17.limit, startAfter = _ref17.startAfter;
              return _context10.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                all_tokens: {
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context10.stop();
          }
        }
      }, _callee10);
    }));
    return function (_x8) {
      return _ref18.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "minter", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee11() {
    return _regenerator["default"].wrap(function _callee11$(_context11) {
      while (1) {
        switch (_context11.prev = _context11.next) {
          case 0:
            return _context11.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              minter: {}
            }));
          case 1:
          case "end":
            return _context11.stop();
        }
      }
    }, _callee11);
  })));
  (0, _defineProperty2["default"])(this, "collectionInfo", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee12() {
    return _regenerator["default"].wrap(function _callee12$(_context12) {
      while (1) {
        switch (_context12.prev = _context12.next) {
          case 0:
            return _context12.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              collection_info: {}
            }));
          case 1:
          case "end":
            return _context12.stop();
        }
      }
    }, _callee12);
  })));
  this.client = client;
  this.contractAddress = contractAddress;
  this.ownerOf = this.ownerOf.bind(this);
  this.approval = this.approval.bind(this);
  this.approvals = this.approvals.bind(this);
  this.allOperators = this.allOperators.bind(this);
  this.numTokens = this.numTokens.bind(this);
  this.contractInfo = this.contractInfo.bind(this);
  this.nftInfo = this.nftInfo.bind(this);
  this.allNftInfo = this.allNftInfo.bind(this);
  this.tokens = this.tokens.bind(this);
  this.allTokens = this.allTokens.bind(this);
  this.minter = this.minter.bind(this);
  this.collectionInfo = this.collectionInfo.bind(this);
});
exports.SG721BaseQueryClient = SG721BaseQueryClient;
