// stargazejs/main/codegen/SgEthAirdrop.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SgEthAirdropQueryClient = exports.SgEthAirdropClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var SgEthAirdropQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function SgEthAirdropQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, SgEthAirdropQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "airdropEligible", /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(_ref) {
      var ethAddress;
      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              ethAddress = _ref.ethAddress;
              return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                airdrop_eligible: {
                  eth_address: ethAddress
                }
              }));
            case 2:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "getMinter", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              get_minter: {}
            }));
          case 1:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  })));
  this.client = client;
  this.contractAddress = contractAddress;
  this.airdropEligible = this.airdropEligible.bind(this);
  this.getMinter = this.getMinter.bind(this);
});
exports.SgEthAirdropQueryClient = SgEthAirdropQueryClient;
var SgEthAirdropClient = /*#__PURE__*/function (_SgEthAirdropQueryCli) {
  (0, _inherits2["default"])(SgEthAirdropClient, _SgEthAirdropQueryCli);
  var _super = _createSuper(SgEthAirdropClient);
  function SgEthAirdropClient(client, sender, contractAddress) {
    var _this2;
    (0, _classCallCheck2["default"])(this, SgEthAirdropClient);
    _this2 = _super.call(this, client, contractAddress);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "client", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sender", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "contractAddress", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "claimAirdrop", /*#__PURE__*/function () {
      var _ref5 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(_ref4) {
        var ethAddress,
          ethSig,
          fee,
          memo,
          _funds,
          _args3 = arguments;
        return _regenerator["default"].wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                ethAddress = _ref4.ethAddress, ethSig = _ref4.ethSig;
                fee = _args3.length > 1 && _args3[1] !== undefined ? _args3[1] : "auto";
                memo = _args3.length > 2 ? _args3[2] : undefined;
                _funds = _args3.length > 3 ? _args3[3] : undefined;
                _context3.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  claim_airdrop: {
                    eth_address: ethAddress,
                    eth_sig: ethSig
                  }
                }, fee, memo, _funds);
              case 6:
                return _context3.abrupt("return", _context3.sent);
              case 7:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3);
      }));
      return function (_x2) {
        return _ref5.apply(this, arguments);
      };
    }());
    _this2.client = client;
    _this2.sender = sender;
    _this2.contractAddress = contractAddress;
    _this2.claimAirdrop = _this2.claimAirdrop.bind((0, _assertThisInitialized2["default"])(_this2));
    return _this2;
  }
  return (0, _createClass2["default"])(SgEthAirdropClient);
}(SgEthAirdropQueryClient);
exports.SgEthAirdropClient = SgEthAirdropClient;
