// stargazejs/main/codegen/Splits.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SplitsQueryClient = exports.SplitsClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var SplitsQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function SplitsQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, SplitsQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "admin", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              admin: {}
            }));
          case 1:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  })));
  (0, _defineProperty2["default"])(this, "group", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              group: {}
            }));
          case 1:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  })));
  (0, _defineProperty2["default"])(this, "member", /*#__PURE__*/function () {
    var _ref4 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(_ref3) {
      var address;
      return _regenerator["default"].wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              address = _ref3.address;
              return _context3.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                member: {
                  address: address
                }
              }));
            case 2:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3);
    }));
    return function (_x) {
      return _ref4.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "listMembers", /*#__PURE__*/function () {
    var _ref6 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(_ref5) {
      var limit, startAfter;
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              limit = _ref5.limit, startAfter = _ref5.startAfter;
              return _context4.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                list_members: {
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4);
    }));
    return function (_x2) {
      return _ref6.apply(this, arguments);
    };
  }());
  this.client = client;
  this.contractAddress = contractAddress;
  this.admin = this.admin.bind(this);
  this.group = this.group.bind(this);
  this.member = this.member.bind(this);
  this.listMembers = this.listMembers.bind(this);
});
exports.SplitsQueryClient = SplitsQueryClient;
var SplitsClient = /*#__PURE__*/function (_SplitsQueryClient) {
  (0, _inherits2["default"])(SplitsClient, _SplitsQueryClient);
  var _super = _createSuper(SplitsClient);
  function SplitsClient(client, sender, contractAddress) {
    var _this2;
    (0, _classCallCheck2["default"])(this, SplitsClient);
    _this2 = _super.call(this, client, contractAddress);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "client", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sender", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "contractAddress", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateAdmin", /*#__PURE__*/function () {
      var _ref8 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(_ref7) {
        var admin,
          fee,
          memo,
          _funds,
          _args5 = arguments;
        return _regenerator["default"].wrap(function _callee5$(_context5) {
          while (1) {
            switch (_context5.prev = _context5.next) {
              case 0:
                admin = _ref7.admin;
                fee = _args5.length > 1 && _args5[1] !== undefined ? _args5[1] : "auto";
                memo = _args5.length > 2 ? _args5[2] : undefined;
                _funds = _args5.length > 3 ? _args5[3] : undefined;
                _context5.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_admin: {
                    admin: admin
                  }
                }, fee, memo, _funds);
              case 6:
                return _context5.abrupt("return", _context5.sent);
              case 7:
              case "end":
                return _context5.stop();
            }
          }
        }, _callee5);
      }));
      return function (_x3) {
        return _ref8.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "distribute", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6() {
      var fee,
        memo,
        _funds,
        _args6 = arguments;
      return _regenerator["default"].wrap(function _callee6$(_context6) {
        while (1) {
          switch (_context6.prev = _context6.next) {
            case 0:
              fee = _args6.length > 0 && _args6[0] !== undefined ? _args6[0] : "auto";
              memo = _args6.length > 1 ? _args6[1] : undefined;
              _funds = _args6.length > 2 ? _args6[2] : undefined;
              _context6.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                distribute: {}
              }, fee, memo, _funds);
            case 5:
              return _context6.abrupt("return", _context6.sent);
            case 6:
            case "end":
              return _context6.stop();
          }
        }
      }, _callee6);
    })));
    _this2.client = client;
    _this2.sender = sender;
    _this2.contractAddress = contractAddress;
    _this2.updateAdmin = _this2.updateAdmin.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.distribute = _this2.distribute.bind((0, _assertThisInitialized2["default"])(_this2));
    return _this2;
  }
  return (0, _createClass2["default"])(SplitsClient);
}(SplitsQueryClient);
exports.SplitsClient = SplitsClient;
