// stargazejs/main/codegen/VendingMinter.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VendingMinterQueryClient = exports.VendingMinterClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var VendingMinterQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function VendingMinterQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, VendingMinterQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "config", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              config: {}
            }));
          case 1:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  })));
  (0, _defineProperty2["default"])(this, "mintableNumTokens", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              mintable_num_tokens: {}
            }));
          case 1:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  })));
  (0, _defineProperty2["default"])(this, "startTime", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3() {
    return _regenerator["default"].wrap(function _callee3$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            return _context3.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              start_time: {}
            }));
          case 1:
          case "end":
            return _context3.stop();
        }
      }
    }, _callee3);
  })));
  (0, _defineProperty2["default"])(this, "mintPrice", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4() {
    return _regenerator["default"].wrap(function _callee4$(_context4) {
      while (1) {
        switch (_context4.prev = _context4.next) {
          case 0:
            return _context4.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              mint_price: {}
            }));
          case 1:
          case "end":
            return _context4.stop();
        }
      }
    }, _callee4);
  })));
  (0, _defineProperty2["default"])(this, "mintCount", /*#__PURE__*/function () {
    var _ref6 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(_ref5) {
      var address;
      return _regenerator["default"].wrap(function _callee5$(_context5) {
        while (1) {
          switch (_context5.prev = _context5.next) {
            case 0:
              address = _ref5.address;
              return _context5.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                mint_count: {
                  address: address
                }
              }));
            case 2:
            case "end":
              return _context5.stop();
          }
        }
      }, _callee5);
    }));
    return function (_x) {
      return _ref6.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "status", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6() {
    return _regenerator["default"].wrap(function _callee6$(_context6) {
      while (1) {
        switch (_context6.prev = _context6.next) {
          case 0:
            return _context6.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              status: {}
            }));
          case 1:
          case "end":
            return _context6.stop();
        }
      }
    }, _callee6);
  })));
  this.client = client;
  this.contractAddress = contractAddress;
  this.config = this.config.bind(this);
  this.mintableNumTokens = this.mintableNumTokens.bind(this);
  this.startTime = this.startTime.bind(this);
  this.mintPrice = this.mintPrice.bind(this);
  this.mintCount = this.mintCount.bind(this);
  this.status = this.status.bind(this);
});
exports.VendingMinterQueryClient = VendingMinterQueryClient;
var VendingMinterClient = /*#__PURE__*/function (_VendingMinterQueryCl) {
  (0, _inherits2["default"])(VendingMinterClient, _VendingMinterQueryCl);
  var _super = _createSuper(VendingMinterClient);
  function VendingMinterClient(client, sender, contractAddress) {
    var _this2;
    (0, _classCallCheck2["default"])(this, VendingMinterClient);
    _this2 = _super.call(this, client, contractAddress);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "client", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sender", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "contractAddress", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "mint", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7() {
      var fee,
        memo,
        _funds,
        _args7 = arguments;
      return _regenerator["default"].wrap(function _callee7$(_context7) {
        while (1) {
          switch (_context7.prev = _context7.next) {
            case 0:
              fee = _args7.length > 0 && _args7[0] !== undefined ? _args7[0] : "auto";
              memo = _args7.length > 1 ? _args7[1] : undefined;
              _funds = _args7.length > 2 ? _args7[2] : undefined;
              _context7.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                mint: {}
              }, fee, memo, _funds);
            case 5:
              return _context7.abrupt("return", _context7.sent);
            case 6:
            case "end":
              return _context7.stop();
          }
        }
      }, _callee7);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "setWhitelist", /*#__PURE__*/function () {
      var _ref10 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee8(_ref9) {
        var whitelist,
          fee,
          memo,
          _funds,
          _args8 = arguments;
        return _regenerator["default"].wrap(function _callee8$(_context8) {
          while (1) {
            switch (_context8.prev = _context8.next) {
              case 0:
                whitelist = _ref9.whitelist;
                fee = _args8.length > 1 && _args8[1] !== undefined ? _args8[1] : "auto";
                memo = _args8.length > 2 ? _args8[2] : undefined;
                _funds = _args8.length > 3 ? _args8[3] : undefined;
                _context8.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  set_whitelist: {
                    whitelist: whitelist
                  }
                }, fee, memo, _funds);
              case 6:
                return _context8.abrupt("return", _context8.sent);
              case 7:
              case "end":
                return _context8.stop();
            }
          }
        }, _callee8);
      }));
      return function (_x2) {
        return _ref10.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "purge", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee9() {
      var fee,
        memo,
        _funds,
        _args9 = arguments;
      return _regenerator["default"].wrap(function _callee9$(_context9) {
        while (1) {
          switch (_context9.prev = _context9.next) {
            case 0:
              fee = _args9.length > 0 && _args9[0] !== undefined ? _args9[0] : "auto";
              memo = _args9.length > 1 ? _args9[1] : undefined;
              _funds = _args9.length > 2 ? _args9[2] : undefined;
              _context9.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                purge: {}
              }, fee, memo, _funds);
            case 5:
              return _context9.abrupt("return", _context9.sent);
            case 6:
            case "end":
              return _context9.stop();
          }
        }
      }, _callee9);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateMintPrice", /*#__PURE__*/function () {
      var _ref13 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee10(_ref12) {
        var price,
          fee,
          memo,
          _funds,
          _args10 = arguments;
        return _regenerator["default"].wrap(function _callee10$(_context10) {
          while (1) {
            switch (_context10.prev = _context10.next) {
              case 0:
                price = _ref12.price;
                fee = _args10.length > 1 && _args10[1] !== undefined ? _args10[1] : "auto";
                memo = _args10.length > 2 ? _args10[2] : undefined;
                _funds = _args10.length > 3 ? _args10[3] : undefined;
                _context10.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_mint_price: {
                    price: price
                  }
                }, fee, memo, _funds);
              case 6:
                return _context10.abrupt("return", _context10.sent);
              case 7:
              case "end":
                return _context10.stop();
            }
          }
        }, _callee10);
      }));
      return function (_x3) {
        return _ref13.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateStartTime", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee11() {
      var fee,
        memo,
        _funds,
        _args11 = arguments;
      return _regenerator["default"].wrap(function _callee11$(_context11) {
        while (1) {
          switch (_context11.prev = _context11.next) {
            case 0:
              fee = _args11.length > 0 && _args11[0] !== undefined ? _args11[0] : "auto";
              memo = _args11.length > 1 ? _args11[1] : undefined;
              _funds = _args11.length > 2 ? _args11[2] : undefined;
              _context11.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                update_start_time: {}
              }, fee, memo, _funds);
            case 5:
              return _context11.abrupt("return", _context11.sent);
            case 6:
            case "end":
              return _context11.stop();
          }
        }
      }, _callee11);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateStartTradingTime", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee12() {
      var fee,
        memo,
        _funds,
        _args12 = arguments;
      return _regenerator["default"].wrap(function _callee12$(_context12) {
        while (1) {
          switch (_context12.prev = _context12.next) {
            case 0:
              fee = _args12.length > 0 && _args12[0] !== undefined ? _args12[0] : "auto";
              memo = _args12.length > 1 ? _args12[1] : undefined;
              _funds = _args12.length > 2 ? _args12[2] : undefined;
              _context12.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                update_start_trading_time: {}
              }, fee, memo, _funds);
            case 5:
              return _context12.abrupt("return", _context12.sent);
            case 6:
            case "end":
              return _context12.stop();
          }
        }
      }, _callee12);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updatePerAddressLimit", /*#__PURE__*/function () {
      var _ref17 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee13(_ref16) {
        var perAddressLimit,
          fee,
          memo,
          _funds,
          _args13 = arguments;
        return _regenerator["default"].wrap(function _callee13$(_context13) {
          while (1) {
            switch (_context13.prev = _context13.next) {
              case 0:
                perAddressLimit = _ref16.perAddressLimit;
                fee = _args13.length > 1 && _args13[1] !== undefined ? _args13[1] : "auto";
                memo = _args13.length > 2 ? _args13[2] : undefined;
                _funds = _args13.length > 3 ? _args13[3] : undefined;
                _context13.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_per_address_limit: {
                    per_address_limit: perAddressLimit
                  }
                }, fee, memo, _funds);
              case 6:
                return _context13.abrupt("return", _context13.sent);
              case 7:
              case "end":
                return _context13.stop();
            }
          }
        }, _callee13);
      }));
      return function (_x4) {
        return _ref17.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "mintTo", /*#__PURE__*/function () {
      var _ref19 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee14(_ref18) {
        var recipient,
          fee,
          memo,
          _funds,
          _args14 = arguments;
        return _regenerator["default"].wrap(function _callee14$(_context14) {
          while (1) {
            switch (_context14.prev = _context14.next) {
              case 0:
                recipient = _ref18.recipient;
                fee = _args14.length > 1 && _args14[1] !== undefined ? _args14[1] : "auto";
                memo = _args14.length > 2 ? _args14[2] : undefined;
                _funds = _args14.length > 3 ? _args14[3] : undefined;
                _context14.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  mint_to: {
                    recipient: recipient
                  }
                }, fee, memo, _funds);
              case 6:
                return _context14.abrupt("return", _context14.sent);
              case 7:
              case "end":
                return _context14.stop();
            }
          }
        }, _callee14);
      }));
      return function (_x5) {
        return _ref19.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "mintFor", /*#__PURE__*/function () {
      var _ref21 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee15(_ref20) {
        var recipient,
          tokenId,
          fee,
          memo,
          _funds,
          _args15 = arguments;
        return _regenerator["default"].wrap(function _callee15$(_context15) {
          while (1) {
            switch (_context15.prev = _context15.next) {
              case 0:
                recipient = _ref20.recipient, tokenId = _ref20.tokenId;
                fee = _args15.length > 1 && _args15[1] !== undefined ? _args15[1] : "auto";
                memo = _args15.length > 2 ? _args15[2] : undefined;
                _funds = _args15.length > 3 ? _args15[3] : undefined;
                _context15.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  mint_for: {
                    recipient: recipient,
                    token_id: tokenId
                  }
                }, fee, memo, _funds);
              case 6:
                return _context15.abrupt("return", _context15.sent);
              case 7:
              case "end":
                return _context15.stop();
            }
          }
        }, _callee15);
      }));
      return function (_x6) {
        return _ref21.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "shuffle", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee16() {
      var fee,
        memo,
        _funds,
        _args16 = arguments;
      return _regenerator["default"].wrap(function _callee16$(_context16) {
        while (1) {
          switch (_context16.prev = _context16.next) {
            case 0:
              fee = _args16.length > 0 && _args16[0] !== undefined ? _args16[0] : "auto";
              memo = _args16.length > 1 ? _args16[1] : undefined;
              _funds = _args16.length > 2 ? _args16[2] : undefined;
              _context16.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                shuffle: {}
              }, fee, memo, _funds);
            case 5:
              return _context16.abrupt("return", _context16.sent);
            case 6:
            case "end":
              return _context16.stop();
          }
        }
      }, _callee16);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "burnRemaining", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee17() {
      var fee,
        memo,
        _funds,
        _args17 = arguments;
      return _regenerator["default"].wrap(function _callee17$(_context17) {
        while (1) {
          switch (_context17.prev = _context17.next) {
            case 0:
              fee = _args17.length > 0 && _args17[0] !== undefined ? _args17[0] : "auto";
              memo = _args17.length > 1 ? _args17[1] : undefined;
              _funds = _args17.length > 2 ? _args17[2] : undefined;
              _context17.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                burn_remaining: {}
              }, fee, memo, _funds);
            case 5:
              return _context17.abrupt("return", _context17.sent);
            case 6:
            case "end":
              return _context17.stop();
          }
        }
      }, _callee17);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateDiscountPrice", /*#__PURE__*/function () {
      var _ref25 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee18(_ref24) {
        var price,
          fee,
          memo,
          _funds,
          _args18 = arguments;
        return _regenerator["default"].wrap(function _callee18$(_context18) {
          while (1) {
            switch (_context18.prev = _context18.next) {
              case 0:
                price = _ref24.price;
                fee = _args18.length > 1 && _args18[1] !== undefined ? _args18[1] : "auto";
                memo = _args18.length > 2 ? _args18[2] : undefined;
                _funds = _args18.length > 3 ? _args18[3] : undefined;
                _context18.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_discount_price: {
                    price: price
                  }
                }, fee, memo, _funds);
              case 6:
                return _context18.abrupt("return", _context18.sent);
              case 7:
              case "end":
                return _context18.stop();
            }
          }
        }, _callee18);
      }));
      return function (_x7) {
        return _ref25.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeDiscountPrice", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee19() {
      var fee,
        memo,
        _funds,
        _args19 = arguments;
      return _regenerator["default"].wrap(function _callee19$(_context19) {
        while (1) {
          switch (_context19.prev = _context19.next) {
            case 0:
              fee = _args19.length > 0 && _args19[0] !== undefined ? _args19[0] : "auto";
              memo = _args19.length > 1 ? _args19[1] : undefined;
              _funds = _args19.length > 2 ? _args19[2] : undefined;
              _context19.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                remove_discount_price: {}
              }, fee, memo, _funds);
            case 5:
              return _context19.abrupt("return", _context19.sent);
            case 6:
            case "end":
              return _context19.stop();
          }
        }
      }, _callee19);
    })));
    _this2.client = client;
    _this2.sender = sender;
    _this2.contractAddress = contractAddress;
    _this2.mint = _this2.mint.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.setWhitelist = _this2.setWhitelist.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.purge = _this2.purge.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateMintPrice = _this2.updateMintPrice.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateStartTime = _this2.updateStartTime.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateStartTradingTime = _this2.updateStartTradingTime.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updatePerAddressLimit = _this2.updatePerAddressLimit.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.mintTo = _this2.mintTo.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.mintFor = _this2.mintFor.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.shuffle = _this2.shuffle.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.burnRemaining = _this2.burnRemaining.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateDiscountPrice = _this2.updateDiscountPrice.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeDiscountPrice = _this2.removeDiscountPrice.bind((0, _assertThisInitialized2["default"])(_this2));
    return _this2;
  }
  return (0, _createClass2["default"])(VendingMinterClient);
}(VendingMinterQueryClient);
exports.VendingMinterClient = VendingMinterClient;
