// stargazejs/main/codegen/WhitelistFlex.client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WhitelistFlexQueryClient = exports.WhitelistFlexClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var WhitelistFlexQueryClient = /*#__PURE__*/(0, _createClass2["default"])(function WhitelistFlexQueryClient(client, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, WhitelistFlexQueryClient);
  (0, _defineProperty2["default"])(this, "client", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "hasStarted", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
    return _regenerator["default"].wrap(function _callee$(_context) {
      while (1) {
        switch (_context.prev = _context.next) {
          case 0:
            return _context.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              has_started: {}
            }));
          case 1:
          case "end":
            return _context.stop();
        }
      }
    }, _callee);
  })));
  (0, _defineProperty2["default"])(this, "hasEnded", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            return _context2.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              has_ended: {}
            }));
          case 1:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  })));
  (0, _defineProperty2["default"])(this, "isActive", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3() {
    return _regenerator["default"].wrap(function _callee3$(_context3) {
      while (1) {
        switch (_context3.prev = _context3.next) {
          case 0:
            return _context3.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              is_active: {}
            }));
          case 1:
          case "end":
            return _context3.stop();
        }
      }
    }, _callee3);
  })));
  (0, _defineProperty2["default"])(this, "members", /*#__PURE__*/function () {
    var _ref5 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(_ref4) {
      var limit, startAfter;
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              limit = _ref4.limit, startAfter = _ref4.startAfter;
              return _context4.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                members: {
                  limit: limit,
                  start_after: startAfter
                }
              }));
            case 2:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4);
    }));
    return function (_x) {
      return _ref5.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "hasMember", /*#__PURE__*/function () {
    var _ref7 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(_ref6) {
      var member;
      return _regenerator["default"].wrap(function _callee5$(_context5) {
        while (1) {
          switch (_context5.prev = _context5.next) {
            case 0:
              member = _ref6.member;
              return _context5.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                has_member: {
                  member: member
                }
              }));
            case 2:
            case "end":
              return _context5.stop();
          }
        }
      }, _callee5);
    }));
    return function (_x2) {
      return _ref7.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "member", /*#__PURE__*/function () {
    var _ref9 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6(_ref8) {
      var member;
      return _regenerator["default"].wrap(function _callee6$(_context6) {
        while (1) {
          switch (_context6.prev = _context6.next) {
            case 0:
              member = _ref8.member;
              return _context6.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                member: {
                  member: member
                }
              }));
            case 2:
            case "end":
              return _context6.stop();
          }
        }
      }, _callee6);
    }));
    return function (_x3) {
      return _ref9.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "config", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7() {
    return _regenerator["default"].wrap(function _callee7$(_context7) {
      while (1) {
        switch (_context7.prev = _context7.next) {
          case 0:
            return _context7.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              config: {}
            }));
          case 1:
          case "end":
            return _context7.stop();
        }
      }
    }, _callee7);
  })));
  (0, _defineProperty2["default"])(this, "adminList", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee8() {
    return _regenerator["default"].wrap(function _callee8$(_context8) {
      while (1) {
        switch (_context8.prev = _context8.next) {
          case 0:
            return _context8.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
              admin_list: {}
            }));
          case 1:
          case "end":
            return _context8.stop();
        }
      }
    }, _callee8);
  })));
  (0, _defineProperty2["default"])(this, "canExecute", /*#__PURE__*/function () {
    var _ref13 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee9(_ref12) {
      var msg, sender;
      return _regenerator["default"].wrap(function _callee9$(_context9) {
        while (1) {
          switch (_context9.prev = _context9.next) {
            case 0:
              msg = _ref12.msg, sender = _ref12.sender;
              return _context9.abrupt("return", _this.client.queryContractSmart(_this.contractAddress, {
                can_execute: {
                  msg: msg,
                  sender: sender
                }
              }));
            case 2:
            case "end":
              return _context9.stop();
          }
        }
      }, _callee9);
    }));
    return function (_x4) {
      return _ref13.apply(this, arguments);
    };
  }());
  this.client = client;
  this.contractAddress = contractAddress;
  this.hasStarted = this.hasStarted.bind(this);
  this.hasEnded = this.hasEnded.bind(this);
  this.isActive = this.isActive.bind(this);
  this.members = this.members.bind(this);
  this.hasMember = this.hasMember.bind(this);
  this.member = this.member.bind(this);
  this.config = this.config.bind(this);
  this.adminList = this.adminList.bind(this);
  this.canExecute = this.canExecute.bind(this);
});
exports.WhitelistFlexQueryClient = WhitelistFlexQueryClient;
var WhitelistFlexClient = /*#__PURE__*/function (_WhitelistFlexQueryCl) {
  (0, _inherits2["default"])(WhitelistFlexClient, _WhitelistFlexQueryCl);
  var _super = _createSuper(WhitelistFlexClient);
  function WhitelistFlexClient(client, sender, contractAddress) {
    var _this2;
    (0, _classCallCheck2["default"])(this, WhitelistFlexClient);
    _this2 = _super.call(this, client, contractAddress);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "client", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "sender", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "contractAddress", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateStartTime", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee10() {
      var fee,
        memo,
        _funds,
        _args10 = arguments;
      return _regenerator["default"].wrap(function _callee10$(_context10) {
        while (1) {
          switch (_context10.prev = _context10.next) {
            case 0:
              fee = _args10.length > 0 && _args10[0] !== undefined ? _args10[0] : "auto";
              memo = _args10.length > 1 ? _args10[1] : undefined;
              _funds = _args10.length > 2 ? _args10[2] : undefined;
              _context10.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                update_start_time: {}
              }, fee, memo, _funds);
            case 5:
              return _context10.abrupt("return", _context10.sent);
            case 6:
            case "end":
              return _context10.stop();
          }
        }
      }, _callee10);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateEndTime", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee11() {
      var fee,
        memo,
        _funds,
        _args11 = arguments;
      return _regenerator["default"].wrap(function _callee11$(_context11) {
        while (1) {
          switch (_context11.prev = _context11.next) {
            case 0:
              fee = _args11.length > 0 && _args11[0] !== undefined ? _args11[0] : "auto";
              memo = _args11.length > 1 ? _args11[1] : undefined;
              _funds = _args11.length > 2 ? _args11[2] : undefined;
              _context11.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                update_end_time: {}
              }, fee, memo, _funds);
            case 5:
              return _context11.abrupt("return", _context11.sent);
            case 6:
            case "end":
              return _context11.stop();
          }
        }
      }, _callee11);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "addMembers", /*#__PURE__*/function () {
      var _ref17 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee12(_ref16) {
        var toAdd,
          fee,
          memo,
          _funds,
          _args12 = arguments;
        return _regenerator["default"].wrap(function _callee12$(_context12) {
          while (1) {
            switch (_context12.prev = _context12.next) {
              case 0:
                toAdd = _ref16.toAdd;
                fee = _args12.length > 1 && _args12[1] !== undefined ? _args12[1] : "auto";
                memo = _args12.length > 2 ? _args12[2] : undefined;
                _funds = _args12.length > 3 ? _args12[3] : undefined;
                _context12.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  add_members: {
                    to_add: toAdd
                  }
                }, fee, memo, _funds);
              case 6:
                return _context12.abrupt("return", _context12.sent);
              case 7:
              case "end":
                return _context12.stop();
            }
          }
        }, _callee12);
      }));
      return function (_x5) {
        return _ref17.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "removeMembers", /*#__PURE__*/function () {
      var _ref19 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee13(_ref18) {
        var toRemove,
          fee,
          memo,
          _funds,
          _args13 = arguments;
        return _regenerator["default"].wrap(function _callee13$(_context13) {
          while (1) {
            switch (_context13.prev = _context13.next) {
              case 0:
                toRemove = _ref18.toRemove;
                fee = _args13.length > 1 && _args13[1] !== undefined ? _args13[1] : "auto";
                memo = _args13.length > 2 ? _args13[2] : undefined;
                _funds = _args13.length > 3 ? _args13[3] : undefined;
                _context13.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  remove_members: {
                    to_remove: toRemove
                  }
                }, fee, memo, _funds);
              case 6:
                return _context13.abrupt("return", _context13.sent);
              case 7:
              case "end":
                return _context13.stop();
            }
          }
        }, _callee13);
      }));
      return function (_x6) {
        return _ref19.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "increaseMemberLimit", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee14() {
      var fee,
        memo,
        _funds,
        _args14 = arguments;
      return _regenerator["default"].wrap(function _callee14$(_context14) {
        while (1) {
          switch (_context14.prev = _context14.next) {
            case 0:
              fee = _args14.length > 0 && _args14[0] !== undefined ? _args14[0] : "auto";
              memo = _args14.length > 1 ? _args14[1] : undefined;
              _funds = _args14.length > 2 ? _args14[2] : undefined;
              _context14.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                increase_member_limit: {}
              }, fee, memo, _funds);
            case 5:
              return _context14.abrupt("return", _context14.sent);
            case 6:
            case "end":
              return _context14.stop();
          }
        }
      }, _callee14);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "updateAdmins", /*#__PURE__*/function () {
      var _ref22 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee15(_ref21) {
        var admins,
          fee,
          memo,
          _funds,
          _args15 = arguments;
        return _regenerator["default"].wrap(function _callee15$(_context15) {
          while (1) {
            switch (_context15.prev = _context15.next) {
              case 0:
                admins = _ref21.admins;
                fee = _args15.length > 1 && _args15[1] !== undefined ? _args15[1] : "auto";
                memo = _args15.length > 2 ? _args15[2] : undefined;
                _funds = _args15.length > 3 ? _args15[3] : undefined;
                _context15.next = 6;
                return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                  update_admins: {
                    admins: admins
                  }
                }, fee, memo, _funds);
              case 6:
                return _context15.abrupt("return", _context15.sent);
              case 7:
              case "end":
                return _context15.stop();
            }
          }
        }, _callee15);
      }));
      return function (_x7) {
        return _ref22.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this2), "freeze", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee16() {
      var fee,
        memo,
        _funds,
        _args16 = arguments;
      return _regenerator["default"].wrap(function _callee16$(_context16) {
        while (1) {
          switch (_context16.prev = _context16.next) {
            case 0:
              fee = _args16.length > 0 && _args16[0] !== undefined ? _args16[0] : "auto";
              memo = _args16.length > 1 ? _args16[1] : undefined;
              _funds = _args16.length > 2 ? _args16[2] : undefined;
              _context16.next = 5;
              return _this2.client.execute(_this2.sender, _this2.contractAddress, {
                freeze: {}
              }, fee, memo, _funds);
            case 5:
              return _context16.abrupt("return", _context16.sent);
            case 6:
            case "end":
              return _context16.stop();
          }
        }
      }, _callee16);
    })));
    _this2.client = client;
    _this2.sender = sender;
    _this2.contractAddress = contractAddress;
    _this2.updateStartTime = _this2.updateStartTime.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateEndTime = _this2.updateEndTime.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.addMembers = _this2.addMembers.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.removeMembers = _this2.removeMembers.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.increaseMemberLimit = _this2.increaseMemberLimit.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.updateAdmins = _this2.updateAdmins.bind((0, _assertThisInitialized2["default"])(_this2));
    _this2.freeze = _this2.freeze.bind((0, _assertThisInitialized2["default"])(_this2));
    return _this2;
  }
  return (0, _createClass2["default"])(WhitelistFlexClient);
}(WhitelistFlexQueryClient);
exports.WhitelistFlexClient = WhitelistFlexClient;
