// stargazejs/main/codegen/WhitelistFlex.message-composer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WhitelistFlexMessageComposer = void 0;
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _tx = require("cosmjs-types/cosmwasm/wasm/v1/tx");
var _encoding = require("@cosmjs/encoding");
/**
* This file was automatically generated by @cosmwasm/ts-codegen@0.30.1.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @cosmwasm/ts-codegen generate command to regenerate this file.
*/var WhitelistFlexMessageComposer = /*#__PURE__*/(0, _createClass2["default"])(function WhitelistFlexMessageComposer(sender, contractAddress) {
  var _this = this;
  (0, _classCallCheck2["default"])(this, WhitelistFlexMessageComposer);
  (0, _defineProperty2["default"])(this, "sender", void 0);
  (0, _defineProperty2["default"])(this, "contractAddress", void 0);
  (0, _defineProperty2["default"])(this, "updateStartTime", function (_funds) {
    return {
      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
      value: _tx.MsgExecuteContract.fromPartial({
        sender: _this.sender,
        contract: _this.contractAddress,
        msg: (0, _encoding.toUtf8)(JSON.stringify({
          update_start_time: {}
        })),
        funds: _funds
      })
    };
  });
  (0, _defineProperty2["default"])(this, "updateEndTime", function (_funds) {
    return {
      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
      value: _tx.MsgExecuteContract.fromPartial({
        sender: _this.sender,
        contract: _this.contractAddress,
        msg: (0, _encoding.toUtf8)(JSON.stringify({
          update_end_time: {}
        })),
        funds: _funds
      })
    };
  });
  (0, _defineProperty2["default"])(this, "addMembers", function (_ref, _funds) {
    var toAdd = _ref.toAdd;
    return {
      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
      value: _tx.MsgExecuteContract.fromPartial({
        sender: _this.sender,
        contract: _this.contractAddress,
        msg: (0, _encoding.toUtf8)(JSON.stringify({
          add_members: {
            to_add: toAdd
          }
        })),
        funds: _funds
      })
    };
  });
  (0, _defineProperty2["default"])(this, "removeMembers", function (_ref2, _funds) {
    var toRemove = _ref2.toRemove;
    return {
      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
      value: _tx.MsgExecuteContract.fromPartial({
        sender: _this.sender,
        contract: _this.contractAddress,
        msg: (0, _encoding.toUtf8)(JSON.stringify({
          remove_members: {
            to_remove: toRemove
          }
        })),
        funds: _funds
      })
    };
  });
  (0, _defineProperty2["default"])(this, "increaseMemberLimit", function (_funds) {
    return {
      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
      value: _tx.MsgExecuteContract.fromPartial({
        sender: _this.sender,
        contract: _this.contractAddress,
        msg: (0, _encoding.toUtf8)(JSON.stringify({
          increase_member_limit: {}
        })),
        funds: _funds
      })
    };
  });
  (0, _defineProperty2["default"])(this, "updateAdmins", function (_ref3, _funds) {
    var admins = _ref3.admins;
    return {
      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
      value: _tx.MsgExecuteContract.fromPartial({
        sender: _this.sender,
        contract: _this.contractAddress,
        msg: (0, _encoding.toUtf8)(JSON.stringify({
          update_admins: {
            admins: admins
          }
        })),
        funds: _funds
      })
    };
  });
  (0, _defineProperty2["default"])(this, "freeze", function (_funds) {
    return {
      typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
      value: _tx.MsgExecuteContract.fromPartial({
        sender: _this.sender,
        contract: _this.contractAddress,
        msg: (0, _encoding.toUtf8)(JSON.stringify({
          freeze: {}
        })),
        funds: _funds
      })
    };
  });
  this.sender = sender;
  this.contractAddress = contractAddress;
  this.updateStartTime = this.updateStartTime.bind(this);
  this.updateEndTime = this.updateEndTime.bind(this);
  this.addMembers = this.addMembers.bind(this);
  this.removeMembers = this.removeMembers.bind(this);
  this.increaseMemberLimit = this.increaseMemberLimit.bind(this);
  this.updateAdmins = this.updateAdmins.bind(this);
  this.freeze = this.freeze.bind(this);
});
exports.WhitelistFlexMessageComposer = WhitelistFlexMessageComposer;
