// @cosmos-kit/core/main/bases/chain-wallet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ChainWalletBase = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _cosmwasmStargate = require("@cosmjs/cosmwasm-stargate");
var _stargate = require("@cosmjs/stargate");
var _tx = require("cosmjs-types/cosmos/tx/v1beta1/tx");
var _nameService = require("../name-service");
var _types = require("../types");
var _utils = require("../utils");
var _wallet = require("./wallet");
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var ChainWalletBase = /*#__PURE__*/function (_WalletBase) {
  (0, _inherits2["default"])(ChainWalletBase, _WalletBase);
  var _super = _createSuper(ChainWalletBase);
  // stands for real `chainIsLazy` considered both `globalIsLazy` and `chainIsLazy` settings

  function ChainWalletBase(walletInfo, chainRecord) {
    var _chainRecord$preferre, _chainRecord$preferre2;
    var _this;
    (0, _classCallCheck2["default"])(this, ChainWalletBase);
    _this = _super.call(this, walletInfo);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "_chainRecord", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "rpcEndpoints", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "restEndpoints", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "_rpcEndpoint", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "_restEndpoint", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "offlineSigner", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "namespace", 'cosmos');
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "isLazy", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getRpcEndpoint", /*#__PURE__*/function () {
      var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(isLazy) {
        var _iterator, _step, endpoint, _this$logger;
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) switch (_context.prev = _context.next) {
            case 0:
              _context.t0 = _this._rpcEndpoint;
              if (!_context.t0) {
                _context.next = 5;
                break;
              }
              _context.next = 4;
              return (0, _utils.isValidEndpoint)(_this._rpcEndpoint, (0, _utils.getIsLazy)(void 0, _this.isLazy, _this._rpcEndpoint.isLazy, isLazy, _this.logger), _this.logger);
            case 4:
              _context.t0 = _context.sent;
            case 5:
              if (!_context.t0) {
                _context.next = 7;
                break;
              }
              return _context.abrupt("return", _this._rpcEndpoint);
            case 7:
              _iterator = _createForOfIteratorHelper(_this.rpcEndpoints || []);
              _context.prev = 8;
              _iterator.s();
            case 10:
              if ((_step = _iterator.n()).done) {
                _context.next = 20;
                break;
              }
              endpoint = _step.value;
              _context.next = 14;
              return (0, _utils.isValidEndpoint)(endpoint, (0, _utils.getIsLazy)(void 0, _this.isLazy, endpoint.isLazy, isLazy, _this.logger), _this.logger);
            case 14:
              if (!_context.sent) {
                _context.next = 18;
                break;
              }
              _this._rpcEndpoint = endpoint;
              (_this$logger = _this.logger) === null || _this$logger === void 0 ? void 0 : _this$logger.debug('Using RPC endpoint ' + endpoint);
              return _context.abrupt("return", endpoint);
            case 18:
              _context.next = 10;
              break;
            case 20:
              _context.next = 25;
              break;
            case 22:
              _context.prev = 22;
              _context.t1 = _context["catch"](8);
              _iterator.e(_context.t1);
            case 25:
              _context.prev = 25;
              _iterator.f();
              return _context.finish(25);
            case 28:
              throw new Error("No valid RPC endpoint for chain ".concat(_this.chainName, " in ").concat(_this.walletName, "!"));
            case 29:
            case "end":
              return _context.stop();
          }
        }, _callee, null, [[8, 22, 25, 28]]);
      }));
      return function (_x) {
        return _ref.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getRestEndpoint", /*#__PURE__*/function () {
      var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(isLazy) {
        var _iterator2, _step2, endpoint, _this$logger2;
        return _regenerator["default"].wrap(function _callee2$(_context2) {
          while (1) switch (_context2.prev = _context2.next) {
            case 0:
              _context2.t0 = _this._restEndpoint;
              if (!_context2.t0) {
                _context2.next = 5;
                break;
              }
              _context2.next = 4;
              return (0, _utils.isValidEndpoint)(_this._restEndpoint, (0, _utils.getIsLazy)(void 0, _this.isLazy, _this._restEndpoint.isLazy, isLazy), _this.logger);
            case 4:
              _context2.t0 = _context2.sent;
            case 5:
              if (!_context2.t0) {
                _context2.next = 7;
                break;
              }
              return _context2.abrupt("return", _this._restEndpoint);
            case 7:
              _iterator2 = _createForOfIteratorHelper(_this.restEndpoints || []);
              _context2.prev = 8;
              _iterator2.s();
            case 10:
              if ((_step2 = _iterator2.n()).done) {
                _context2.next = 20;
                break;
              }
              endpoint = _step2.value;
              _context2.next = 14;
              return (0, _utils.isValidEndpoint)(endpoint, (0, _utils.getIsLazy)(void 0, _this.isLazy, endpoint.isLazy, isLazy), _this.logger);
            case 14:
              if (!_context2.sent) {
                _context2.next = 18;
                break;
              }
              _this._restEndpoint = endpoint;
              (_this$logger2 = _this.logger) === null || _this$logger2 === void 0 ? void 0 : _this$logger2.debug('Using REST endpoint ' + endpoint);
              return _context2.abrupt("return", endpoint);
            case 18:
              _context2.next = 10;
              break;
            case 20:
              _context2.next = 25;
              break;
            case 22:
              _context2.prev = 22;
              _context2.t1 = _context2["catch"](8);
              _iterator2.e(_context2.t1);
            case 25:
              _context2.prev = 25;
              _iterator2.f();
              return _context2.finish(25);
            case 28:
              throw new Error("No valid Rest endpoint for chain ".concat(_this.chainName, " in ").concat(_this.walletName, "!"));
            case 29:
            case "end":
              return _context2.stop();
          }
        }, _callee2, null, [[8, 22, 25, 28]]);
      }));
      return function (_x2) {
        return _ref2.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getStargateClient", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3() {
      var rpcEndpoint;
      return _regenerator["default"].wrap(function _callee3$(_context3) {
        while (1) switch (_context3.prev = _context3.next) {
          case 0:
            _context3.next = 2;
            return _this.getRpcEndpoint();
          case 2:
            rpcEndpoint = _context3.sent;
            return _context3.abrupt("return", _stargate.StargateClient.connect(rpcEndpoint, _this.stargateOptions));
          case 4:
          case "end":
            return _context3.stop();
        }
      }, _callee3);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getCosmWasmClient", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4() {
      var rpcEndpoint;
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) switch (_context4.prev = _context4.next) {
          case 0:
            _context4.next = 2;
            return _this.getRpcEndpoint();
          case 2:
            rpcEndpoint = _context4.sent;
            return _context4.abrupt("return", _cosmwasmStargate.CosmWasmClient.connect(rpcEndpoint));
          case 4:
          case "end":
            return _context4.stop();
        }
      }, _callee4);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getNameService", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5() {
      var client, registry;
      return _regenerator["default"].wrap(function _callee5$(_context5) {
        while (1) switch (_context5.prev = _context5.next) {
          case 0:
            _context5.next = 2;
            return _this.getCosmWasmClient();
          case 2:
            client = _context5.sent;
            registry = (0, _utils.getNameServiceRegistryFromChainName)(_this.chainName);
            return _context5.abrupt("return", new _nameService.NameService(client, registry));
          case 5:
          case "end":
            return _context5.stop();
        }
      }, _callee5);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getSigningStargateClient", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6() {
      var rpcEndpoint;
      return _regenerator["default"].wrap(function _callee6$(_context6) {
        while (1) switch (_context6.prev = _context6.next) {
          case 0:
            _context6.next = 2;
            return _this.getRpcEndpoint();
          case 2:
            rpcEndpoint = _context6.sent;
            if (_this.offlineSigner) {
              _context6.next = 6;
              break;
            }
            _context6.next = 6;
            return _this.initOfflineSigner();
          case 6:
            return _context6.abrupt("return", _stargate.SigningStargateClient.connectWithSigner(rpcEndpoint, _this.offlineSigner, _this.signingStargateOptions));
          case 7:
          case "end":
            return _context6.stop();
        }
      }, _callee6);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getSigningCosmWasmClient", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7() {
      var rpcEndpoint;
      return _regenerator["default"].wrap(function _callee7$(_context7) {
        while (1) switch (_context7.prev = _context7.next) {
          case 0:
            _context7.next = 2;
            return _this.getRpcEndpoint();
          case 2:
            rpcEndpoint = _context7.sent;
            if (_this.offlineSigner) {
              _context7.next = 6;
              break;
            }
            _context7.next = 6;
            return _this.initOfflineSigner();
          case 6:
            return _context7.abrupt("return", _cosmwasmStargate.SigningCosmWasmClient.connectWithSigner(rpcEndpoint, _this.offlineSigner, _this.signingCosmwasmOptions));
          case 7:
          case "end":
            return _context7.stop();
        }
      }, _callee7);
    })));
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getSigningClient", /*#__PURE__*/function () {
      var _ref8 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee8(type) {
        return _regenerator["default"].wrap(function _callee8$(_context8) {
          while (1) switch (_context8.prev = _context8.next) {
            case 0:
              _context8.t0 = type;
              _context8.next = _context8.t0 === 'stargate' ? 3 : _context8.t0 === 'cosmwasm' ? 6 : 9;
              break;
            case 3:
              _context8.next = 5;
              return _this.getSigningStargateClient();
            case 5:
              return _context8.abrupt("return", _context8.sent);
            case 6:
              _context8.next = 8;
              return _this.getSigningCosmWasmClient();
            case 8:
              return _context8.abrupt("return", _context8.sent);
            case 9:
              return _context8.abrupt("return", _this.getSigningStargateClient());
            case 10:
            case "end":
              return _context8.stop();
          }
        }, _callee8);
      }));
      return function (_x3) {
        return _ref8.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "estimateFee", /*#__PURE__*/function () {
      var _ref9 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee9(messages, type, memo, multiplier) {
        var _this$signingStargate, _this$signingCosmwasm, _this$signingStargate2;
        var gasPrice, client, gasEstimation;
        return _regenerator["default"].wrap(function _callee9$(_context9) {
          while (1) switch (_context9.prev = _context9.next) {
            case 0:
              if (_this.address) {
                _context9.next = 2;
                break;
              }
              throw new Error('Address is required to estimate fee. Try connect to fetch address.');
            case 2:
              _context9.t0 = type;
              _context9.next = _context9.t0 === 'stargate' ? 5 : _context9.t0 === 'cosmwasm' ? 7 : 9;
              break;
            case 5:
              gasPrice = (_this$signingStargate = _this.signingStargateOptions) === null || _this$signingStargate === void 0 ? void 0 : _this$signingStargate.gasPrice;
              return _context9.abrupt("break", 11);
            case 7:
              gasPrice = (_this$signingCosmwasm = _this.signingCosmwasmOptions) === null || _this$signingCosmwasm === void 0 ? void 0 : _this$signingCosmwasm.gasPrice;
              return _context9.abrupt("break", 11);
            case 9:
              gasPrice = (_this$signingStargate2 = _this.signingStargateOptions) === null || _this$signingStargate2 === void 0 ? void 0 : _this$signingStargate2.gasPrice;
              return _context9.abrupt("break", 11);
            case 11:
              if (gasPrice) {
                _context9.next = 13;
                break;
              }
              throw new Error('Gas price must be set in the client options when auto gas is used.');
            case 13:
              _context9.next = 15;
              return _this.getSigningClient(type);
            case 15:
              client = _context9.sent;
              _context9.next = 18;
              return client.simulate(_this.address, messages, memo);
            case 18:
              gasEstimation = _context9.sent;
              return _context9.abrupt("return", (0, _stargate.calculateFee)(Math.round(gasEstimation * (multiplier || 1.3)), gasPrice));
            case 20:
            case "end":
              return _context9.stop();
          }
        }, _callee9);
      }));
      return function (_x4, _x5, _x6, _x7) {
        return _ref9.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "sign", /*#__PURE__*/function () {
      var _ref10 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee10(messages, fee, memo, type) {
        var client, usedFee;
        return _regenerator["default"].wrap(function _callee10$(_context10) {
          while (1) switch (_context10.prev = _context10.next) {
            case 0:
              if (_this.address) {
                _context10.next = 2;
                break;
              }
              throw new Error('Address is required to estimate fee. Try connect to fetch address.');
            case 2:
              _context10.next = 4;
              return _this.getSigningClient(type);
            case 4:
              client = _context10.sent;
              if (!(typeof fee === 'undefined' || typeof fee === 'number')) {
                _context10.next = 11;
                break;
              }
              _context10.next = 8;
              return _this.estimateFee(messages, type, memo, fee);
            case 8:
              usedFee = _context10.sent;
              _context10.next = 12;
              break;
            case 11:
              usedFee = fee;
            case 12:
              _context10.next = 14;
              return client.sign(_this.address, messages, usedFee, memo || '');
            case 14:
              return _context10.abrupt("return", _context10.sent);
            case 15:
            case "end":
              return _context10.stop();
          }
        }, _callee10);
      }));
      return function (_x8, _x9, _x10, _x11) {
        return _ref10.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "broadcast", /*#__PURE__*/function () {
      var _ref11 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee11(signedMessages, type) {
        var _this$signingStargate3, _this$signingStargate4, _this$signingCosmwasm2, _this$signingCosmwasm3, _this$signingStargate5, _this$signingStargate6;
        var client, txBytes, timeoutMs, pollIntervalMs;
        return _regenerator["default"].wrap(function _callee11$(_context11) {
          while (1) switch (_context11.prev = _context11.next) {
            case 0:
              _context11.next = 2;
              return _this.getSigningClient(type);
            case 2:
              client = _context11.sent;
              txBytes = _tx.TxRaw.encode(signedMessages).finish();
              _context11.t0 = type;
              _context11.next = _context11.t0 === 'stargate' ? 7 : _context11.t0 === 'cosmwasm' ? 10 : 13;
              break;
            case 7:
              timeoutMs = (_this$signingStargate3 = _this.signingStargateOptions) === null || _this$signingStargate3 === void 0 ? void 0 : _this$signingStargate3.broadcastTimeoutMs;
              pollIntervalMs = (_this$signingStargate4 = _this.signingStargateOptions) === null || _this$signingStargate4 === void 0 ? void 0 : _this$signingStargate4.broadcastPollIntervalMs;
              return _context11.abrupt("break", 16);
            case 10:
              timeoutMs = (_this$signingCosmwasm2 = _this.signingCosmwasmOptions) === null || _this$signingCosmwasm2 === void 0 ? void 0 : _this$signingCosmwasm2.broadcastTimeoutMs;
              pollIntervalMs = (_this$signingCosmwasm3 = _this.signingCosmwasmOptions) === null || _this$signingCosmwasm3 === void 0 ? void 0 : _this$signingCosmwasm3.broadcastPollIntervalMs;
              return _context11.abrupt("break", 16);
            case 13:
              timeoutMs = (_this$signingStargate5 = _this.signingStargateOptions) === null || _this$signingStargate5 === void 0 ? void 0 : _this$signingStargate5.broadcastTimeoutMs;
              pollIntervalMs = (_this$signingStargate6 = _this.signingStargateOptions) === null || _this$signingStargate6 === void 0 ? void 0 : _this$signingStargate6.broadcastPollIntervalMs;
              return _context11.abrupt("break", 16);
            case 16:
              return _context11.abrupt("return", client.broadcastTx(txBytes, timeoutMs, pollIntervalMs));
            case 17:
            case "end":
              return _context11.stop();
          }
        }, _callee11);
      }));
      return function (_x12, _x13) {
        return _ref11.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "signAndBroadcast", /*#__PURE__*/function () {
      var _ref12 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee12(messages, fee, memo, type) {
        var signedMessages;
        return _regenerator["default"].wrap(function _callee12$(_context12) {
          while (1) switch (_context12.prev = _context12.next) {
            case 0:
              _context12.next = 2;
              return _this.sign(messages, fee, memo, type);
            case 2:
              signedMessages = _context12.sent;
              return _context12.abrupt("return", _this.broadcast(signedMessages, type));
            case 4:
            case "end":
              return _context12.stop();
          }
        }, _callee12);
      }));
      return function (_x14, _x15, _x16, _x17) {
        return _ref12.apply(this, arguments);
      };
    }());
    _this._chainRecord = chainRecord;
    _this.rpcEndpoints = (_chainRecord$preferre = chainRecord.preferredEndpoints) === null || _chainRecord$preferre === void 0 ? void 0 : _chainRecord$preferre.rpc;
    _this.restEndpoints = (_chainRecord$preferre2 = chainRecord.preferredEndpoints) === null || _chainRecord$preferre2 === void 0 ? void 0 : _chainRecord$preferre2.rest;
    return _this;
  }
  (0, _createClass2["default"])(ChainWalletBase, [{
    key: "chainRecord",
    get: function get() {
      return this._chainRecord;
    }
  }, {
    key: "chainName",
    get: function get() {
      return this.chainRecord.name;
    }
  }, {
    key: "chainLogoUrl",
    get: function get() {
      var _this$chainRecord$ass, _this$chainRecord$ass2, _this$chainRecord$ass3, _this$chainRecord$ass4, _this$chainRecord$ass5, _this$chainRecord$ass6;
      return (
        // until chain_registry fix this
        // this.chainInfo.chain.logo_URIs?.svg ||
        // this.chainInfo.chain.logo_URIs?.png ||
        // this.chainInfo.chain.logo_URIs?.jpeg ||
        ((_this$chainRecord$ass = this.chainRecord.assetList) === null || _this$chainRecord$ass === void 0 ? void 0 : (_this$chainRecord$ass2 = _this$chainRecord$ass.assets[0]) === null || _this$chainRecord$ass2 === void 0 ? void 0 : (_this$chainRecord$ass3 = _this$chainRecord$ass2.logo_URIs) === null || _this$chainRecord$ass3 === void 0 ? void 0 : _this$chainRecord$ass3.svg) || ((_this$chainRecord$ass4 = this.chainRecord.assetList) === null || _this$chainRecord$ass4 === void 0 ? void 0 : (_this$chainRecord$ass5 = _this$chainRecord$ass4.assets[0]) === null || _this$chainRecord$ass5 === void 0 ? void 0 : (_this$chainRecord$ass6 = _this$chainRecord$ass5.logo_URIs) === null || _this$chainRecord$ass6 === void 0 ? void 0 : _this$chainRecord$ass6.png) || undefined
      );
    }
  }, {
    key: "stargateOptions",
    get: function get() {
      var _this$chainRecord$cli;
      return (_this$chainRecord$cli = this.chainRecord.clientOptions) === null || _this$chainRecord$cli === void 0 ? void 0 : _this$chainRecord$cli.stargate;
    }
  }, {
    key: "signingStargateOptions",
    get: function get() {
      var _this$chainRecord$cli2, _this$chainRecord$cli3;
      return ((_this$chainRecord$cli2 = this.chainRecord.clientOptions) === null || _this$chainRecord$cli2 === void 0 ? void 0 : _this$chainRecord$cli2.signingStargate) || ((_this$chainRecord$cli3 = this.chainRecord.clientOptions) === null || _this$chainRecord$cli3 === void 0 ? void 0 : _this$chainRecord$cli3.stargate);
    }
  }, {
    key: "signingCosmwasmOptions",
    get: function get() {
      var _this$chainRecord$cli4;
      return (_this$chainRecord$cli4 = this.chainRecord.clientOptions) === null || _this$chainRecord$cli4 === void 0 ? void 0 : _this$chainRecord$cli4.signingCosmwasm;
    }
  }, {
    key: "preferredSignType",
    get: function get() {
      var _this$chainRecord$cli5;
      return ((_this$chainRecord$cli5 = this.chainRecord.clientOptions) === null || _this$chainRecord$cli5 === void 0 ? void 0 : _this$chainRecord$cli5.preferredSignType) || 'amino';
    }
  }, {
    key: "chain",
    get: function get() {
      return this.chainRecord.chain;
    }
  }, {
    key: "assets",
    get: function get() {
      var _this$chainRecord$ass7;
      return (_this$chainRecord$ass7 = this.chainRecord.assetList) === null || _this$chainRecord$ass7 === void 0 ? void 0 : _this$chainRecord$ass7.assets;
    }
  }, {
    key: "assetList",
    get: function get() {
      return this.chainRecord.assetList;
    }
  }, {
    key: "chainId",
    get: function get() {
      var _this$chain;
      return (_this$chain = this.chain) === null || _this$chain === void 0 ? void 0 : _this$chain.chain_id;
    }
  }, {
    key: "cosmwasmEnabled",
    get: function get() {
      var _this$chain2, _this$chain2$codebase;
      return (_this$chain2 = this.chain) === null || _this$chain2 === void 0 ? void 0 : (_this$chain2$codebase = _this$chain2.codebase) === null || _this$chain2$codebase === void 0 ? void 0 : _this$chain2$codebase.cosmwasm_enabled;
    }
  }, {
    key: "username",
    get: function get() {
      var _this$data;
      return (_this$data = this.data) === null || _this$data === void 0 ? void 0 : _this$data.username;
    }
  }, {
    key: "address",
    get: function get() {
      var _this$data2;
      return (_this$data2 = this.data) === null || _this$data2 === void 0 ? void 0 : _this$data2.address;
    }
  }, {
    key: "setData",
    value: function setData(data) {
      var _this$actions,
        _this$actions$data,
        _this2 = this,
        _window,
        _this$session;
      this._mutable.data = data;
      (_this$actions = this.actions) === null || _this$actions === void 0 ? void 0 : (_this$actions$data = _this$actions.data) === null || _this$actions$data === void 0 ? void 0 : _this$actions$data.call(_this$actions, data);
      var accountsStr = window.localStorage.getItem('cosmos-kit@1:core//accounts');
      var accounts = accountsStr ? JSON.parse(accountsStr) : [];
      if (typeof data === 'undefined') {
        accounts = accounts.filter(function (a) {
          return a.chainId !== _this2.chainId || a.namespace !== _this2.namespace;
        });
      } else {
        accounts = accounts.filter(function (a) {
          return a.chainId !== _this2.chainId || a.namespace !== _this2.namespace;
        });
        accounts.push(data);
      }
      (_window = window) === null || _window === void 0 ? void 0 : _window.localStorage.setItem('cosmos-kit@1:core//accounts', JSON.stringify(accounts));
      (_this$session = this.session) === null || _this$session === void 0 ? void 0 : _this$session.update();
    }
  }, {
    key: "initClient",
    value: function initClient(_options) {
      throw new Error('initClient not implemented');
    }
  }, {
    key: "update",
    value: function () {
      var _update = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee13() {
        var _this$client$connect, _this$client, account, _this$logger3, _this$logger4, _window2;
        return _regenerator["default"].wrap(function _callee13$(_context13) {
          while (1) switch (_context13.prev = _context13.next) {
            case 0:
              this.setState(_types.State.Pending);
              this.setMessage(void 0);
              _context13.prev = 2;
              _context13.next = 5;
              return (_this$client$connect = (_this$client = this.client).connect) === null || _this$client$connect === void 0 ? void 0 : _this$client$connect.call(_this$client, this.chainId);
            case 5:
              _context13.prev = 5;
              (_this$logger3 = this.logger) === null || _this$logger3 === void 0 ? void 0 : _this$logger3.debug("Fetching ".concat(this.walletName, " ").concat(this.chainId, " account."));
              _context13.next = 9;
              return this.client.getSimpleAccount(this.chainId);
            case 9:
              account = _context13.sent;
              _context13.next = 26;
              break;
            case 12:
              _context13.prev = 12;
              _context13.t0 = _context13["catch"](5);
              if (!this.rejectMatched(_context13.t0)) {
                _context13.next = 17;
                break;
              }
              this.setRejected();
              return _context13.abrupt("return");
            case 17:
              if (!this.client.addChain) {
                _context13.next = 25;
                break;
              }
              _context13.next = 20;
              return this.client.addChain(this.chainRecord);
            case 20:
              _context13.next = 22;
              return this.client.getSimpleAccount(this.chainId);
            case 22:
              account = _context13.sent;
              _context13.next = 26;
              break;
            case 25:
              throw _context13.t0;
            case 26:
              this.setData(account);
              this.setState(_types.State.Done);
              this.setMessage(void 0);
              _context13.next = 35;
              break;
            case 31:
              _context13.prev = 31;
              _context13.t1 = _context13["catch"](2);
              (_this$logger4 = this.logger) === null || _this$logger4 === void 0 ? void 0 : _this$logger4.error(_context13.t1);
              if (_context13.t1 && this.rejectMatched(_context13.t1)) {
                this.setRejected();
              } else {
                this.setError(_context13.t1);
              }
            case 35:
              if (!this.isWalletRejected) {
                (_window2 = window) === null || _window2 === void 0 ? void 0 : _window2.localStorage.setItem('cosmos-kit@1:core//current-wallet', this.walletName);
              }
            case 36:
            case "end":
              return _context13.stop();
          }
        }, _callee13, this, [[2, 31], [5, 12]]);
      }));
      function update() {
        return _update.apply(this, arguments);
      }
      return update;
    }()
  }, {
    key: "initOfflineSigner",
    value: function () {
      var _initOfflineSigner = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee14() {
        return _regenerator["default"].wrap(function _callee14$(_context14) {
          while (1) switch (_context14.prev = _context14.next) {
            case 0:
              if (!(typeof this.client === 'undefined')) {
                _context14.next = 2;
                break;
              }
              throw new Error('WalletClient is not initialized');
            case 2:
              _context14.next = 4;
              return this.client.getOfflineSigner(this.chainId, this.preferredSignType);
            case 4:
              this.offlineSigner = _context14.sent;
            case 5:
            case "end":
              return _context14.stop();
          }
        }, _callee14, this);
      }));
      function initOfflineSigner() {
        return _initOfflineSigner.apply(this, arguments);
      }
      return initOfflineSigner;
    }()
  }]);
  return ChainWalletBase;
}(_wallet.WalletBase);
exports.ChainWalletBase = ChainWalletBase;
