// @cosmos-kit/core/main/bases/main-wallet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MainWalletBase = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _types = require("../types");
var _wallet = require("./wallet");
var _events = _interopRequireDefault(require("events"));
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var MainWalletBase = /*#__PURE__*/function (_WalletBase) {
  (0, _inherits2["default"])(MainWalletBase, _WalletBase);
  var _super = _createSuper(MainWalletBase);
  function MainWalletBase(walletInfo, ChainWallet) {
    var _this;
    (0, _classCallCheck2["default"])(this, MainWalletBase);
    _this = _super.call(this, walletInfo);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "_chainWalletMap", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "preferredEndpoints", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "ChainWallet", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getChainWallet", function (chainName) {
      var _this$chainWalletMap;
      return (_this$chainWalletMap = _this.chainWalletMap) === null || _this$chainWalletMap === void 0 ? void 0 : _this$chainWalletMap.get(chainName);
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getChainWalletList", function () {
      var activeOnly = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : true;
      var allChainWallets = Array.from(_this.chainWalletMap.values());
      return activeOnly ? allChainWallets.filter(function (w) {
        return w.isActive;
      }) : allChainWallets;
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "getGlobalStatusAndMessage", function () {
      var activeOnly = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : true;
      var chainWalletList = _this.getChainWalletList(activeOnly);
      var wallet = chainWalletList.find(function (w) {
        return w.isWalletNotExist;
      });
      if (wallet) return [wallet.walletStatus, wallet.message];
      wallet = chainWalletList.find(function (w) {
        return w.isWalletConnecting;
      });
      if (wallet) return [_types.WalletStatus.Connecting, void 0];
      wallet = chainWalletList.find(function (w) {
        return w.isWalletDisconnected;
      });
      if (wallet) {
        return [_types.WalletStatus.Disconnected, 'Exist disconnected wallets'];
      }
      wallet = chainWalletList.find(function (w) {
        return w.isError || w.isWalletRejected;
      });
      if (wallet) return [wallet.walletStatus, wallet.message];
      return [_types.WalletStatus.Connected, void 0];
    });
    _this.ChainWallet = ChainWallet;
    _this.emitter = new _events["default"]();
    _this.emitter.on('broadcast_client', function (client) {
      var _this$chainWalletMap2;
      (_this$chainWalletMap2 = _this.chainWalletMap) === null || _this$chainWalletMap2 === void 0 ? void 0 : _this$chainWalletMap2.forEach(function (chainWallet) {
        chainWallet.initClientDone(client);
      });
    });
    _this.emitter.on('broadcast_env', function (env) {
      var _this$chainWalletMap3;
      (_this$chainWalletMap3 = _this.chainWalletMap) === null || _this$chainWalletMap3 === void 0 ? void 0 : _this$chainWalletMap3.forEach(function (chainWallet) {
        chainWallet.setEnv(env);
      });
    });
    _this.emitter.on('sync_connect', function (chainName) {
      _this.connectAll(true, chainName);
      _this.activate();
    });
    _this.emitter.on('sync_disconnect', function (chainName) {
      _this.disconnectAll(true, chainName);
      _this.inactivate();
    });
    _this.emitter.on('reset', function (chainIds) {
      chainIds.forEach(function (chainId) {
        var _Array$from$find;
        return (_Array$from$find = Array.from(_this.chainWalletMap.values()).find(function (cw) {
          return cw.chainId === chainId;
        })) === null || _Array$from$find === void 0 ? void 0 : _Array$from$find.reset();
      });
    });
    return _this;
  }
  (0, _createClass2["default"])(MainWalletBase, [{
    key: "onSetChainsDone",
    value: function onSetChainsDone() {}
  }, {
    key: "setChains",
    value: function setChains(chains) {
      var _this2 = this;
      var overwrite = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : true;
      if (overwrite || !this._chainWalletMap) {
        this._chainWalletMap = new Map();
      }
      chains.forEach(function (chain) {
        var _chain$preferredEndpo, _this2$preferredEndpo, _this2$preferredEndpo2, _chain$chain, _chain$chain$apis, _chain$chain$apis$rpc, _chain$preferredEndpo2, _this2$preferredEndpo3, _this2$preferredEndpo4, _chain$chain2, _chain$chain2$apis, _chain$chain2$apis$re, _chain$preferredEndpo3;
        chain.preferredEndpoints = _objectSpread(_objectSpread({}, chain.preferredEndpoints), {}, {
          rpc: [].concat((0, _toConsumableArray2["default"])(((_chain$preferredEndpo = chain.preferredEndpoints) === null || _chain$preferredEndpo === void 0 ? void 0 : _chain$preferredEndpo.rpc) || []), (0, _toConsumableArray2["default"])(((_this2$preferredEndpo = _this2.preferredEndpoints) === null || _this2$preferredEndpo === void 0 ? void 0 : (_this2$preferredEndpo2 = _this2$preferredEndpo[chain.name]) === null || _this2$preferredEndpo2 === void 0 ? void 0 : _this2$preferredEndpo2.rpc) || []), ["https://rpc.cosmos.directory/".concat(chain.name)], (0, _toConsumableArray2["default"])(((_chain$chain = chain.chain) === null || _chain$chain === void 0 ? void 0 : (_chain$chain$apis = _chain$chain.apis) === null || _chain$chain$apis === void 0 ? void 0 : (_chain$chain$apis$rpc = _chain$chain$apis.rpc) === null || _chain$chain$apis$rpc === void 0 ? void 0 : _chain$chain$apis$rpc.map(function (e) {
            return e.address;
          })) || [])),
          rest: [].concat((0, _toConsumableArray2["default"])(((_chain$preferredEndpo2 = chain.preferredEndpoints) === null || _chain$preferredEndpo2 === void 0 ? void 0 : _chain$preferredEndpo2.rest) || []), (0, _toConsumableArray2["default"])(((_this2$preferredEndpo3 = _this2.preferredEndpoints) === null || _this2$preferredEndpo3 === void 0 ? void 0 : (_this2$preferredEndpo4 = _this2$preferredEndpo3[chain.name]) === null || _this2$preferredEndpo4 === void 0 ? void 0 : _this2$preferredEndpo4.rest) || []), ["https://rest.cosmos.directory/".concat(chain.name)], (0, _toConsumableArray2["default"])(((_chain$chain2 = chain.chain) === null || _chain$chain2 === void 0 ? void 0 : (_chain$chain2$apis = _chain$chain2.apis) === null || _chain$chain2$apis === void 0 ? void 0 : (_chain$chain2$apis$re = _chain$chain2$apis.rest) === null || _chain$chain2$apis$re === void 0 ? void 0 : _chain$chain2$apis$re.map(function (e) {
            return e.address;
          })) || []))
        });
        var chainWallet = new _this2.ChainWallet(_this2.walletInfo, chain);
        chainWallet.emitter = _this2.emitter;
        chainWallet.logger = _this2.logger;
        chainWallet.session = _this2.session;
        chainWallet.walletConnectOptions = _this2.walletConnectOptions;
        chainWallet.initClient = _this2.initClient;
        chainWallet.isLazy = (_chain$preferredEndpo3 = chain.preferredEndpoints) === null || _chain$preferredEndpo3 === void 0 ? void 0 : _chain$preferredEndpo3.isLazy;
        _this2._chainWalletMap.set(chain.name, chainWallet);
      });
      this.onSetChainsDone();
    }
  }, {
    key: "chainWalletMap",
    get: function get() {
      return this._chainWalletMap;
    }
  }, {
    key: "update",
    value: function () {
      var _update = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee() {
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) switch (_context.prev = _context.next) {
            case 0:
            case "end":
              return _context.stop();
          }
        }, _callee);
      }));
      function update() {
        return _update.apply(this, arguments);
      }
      return update;
    }()
  }, {
    key: "connectAll",
    value: function () {
      var _connectAll = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
        var activeOnly,
          exclude,
          chainWalletList,
          _iterator,
          _step,
          w,
          _args2 = arguments;
        return _regenerator["default"].wrap(function _callee2$(_context2) {
          while (1) switch (_context2.prev = _context2.next) {
            case 0:
              activeOnly = _args2.length > 0 && _args2[0] !== undefined ? _args2[0] : true;
              exclude = _args2.length > 1 ? _args2[1] : undefined;
              chainWalletList = this.getChainWalletList(activeOnly);
              _iterator = _createForOfIteratorHelper(chainWalletList);
              _context2.prev = 4;
              _iterator.s();
            case 6:
              if ((_step = _iterator.n()).done) {
                _context2.next = 13;
                break;
              }
              w = _step.value;
              if (!(w.chainName !== exclude)) {
                _context2.next = 11;
                break;
              }
              _context2.next = 11;
              return w.connect();
            case 11:
              _context2.next = 6;
              break;
            case 13:
              _context2.next = 18;
              break;
            case 15:
              _context2.prev = 15;
              _context2.t0 = _context2["catch"](4);
              _iterator.e(_context2.t0);
            case 18:
              _context2.prev = 18;
              _iterator.f();
              return _context2.finish(18);
            case 21:
            case "end":
              return _context2.stop();
          }
        }, _callee2, this, [[4, 15, 18, 21]]);
      }));
      function connectAll() {
        return _connectAll.apply(this, arguments);
      }
      return connectAll;
    }()
  }, {
    key: "disconnectAll",
    value: function () {
      var _disconnectAll = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3() {
        var activeOnly,
          exclude,
          chainWalletList,
          _iterator2,
          _step2,
          w,
          _args3 = arguments;
        return _regenerator["default"].wrap(function _callee3$(_context3) {
          while (1) switch (_context3.prev = _context3.next) {
            case 0:
              activeOnly = _args3.length > 0 && _args3[0] !== undefined ? _args3[0] : true;
              exclude = _args3.length > 1 ? _args3[1] : undefined;
              chainWalletList = this.getChainWalletList(activeOnly);
              _iterator2 = _createForOfIteratorHelper(chainWalletList);
              _context3.prev = 4;
              _iterator2.s();
            case 6:
              if ((_step2 = _iterator2.n()).done) {
                _context3.next = 13;
                break;
              }
              w = _step2.value;
              if (!(w.chainName !== exclude)) {
                _context3.next = 11;
                break;
              }
              _context3.next = 11;
              return w.disconnect();
            case 11:
              _context3.next = 6;
              break;
            case 13:
              _context3.next = 18;
              break;
            case 15:
              _context3.prev = 15;
              _context3.t0 = _context3["catch"](4);
              _iterator2.e(_context3.t0);
            case 18:
              _context3.prev = 18;
              _iterator2.f();
              return _context3.finish(18);
            case 21:
            case "end":
              return _context3.stop();
          }
        }, _callee3, this, [[4, 15, 18, 21]]);
      }));
      function disconnectAll() {
        return _disconnectAll.apply(this, arguments);
      }
      return disconnectAll;
    }()
  }]);
  return MainWalletBase;
}(_wallet.WalletBase);
exports.MainWalletBase = MainWalletBase;
