// @cosmos-kit/core/main/bases/wallet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WalletBase = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _types = require("../types");
var _utils = require("../utils");
var _state = require("./state");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2["default"])(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2["default"])(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2["default"])(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var WalletBase = /*#__PURE__*/function (_StateBase) {
  (0, _inherits2["default"])(WalletBase, _StateBase);
  var _super = _createSuper(WalletBase);
  function WalletBase(walletInfo) {
    var _this;
    (0, _classCallCheck2["default"])(this, WalletBase);
    _this = _super.call(this);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "clientMutable", {
      state: _types.State.Init
    });
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "emitter", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "_walletInfo", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "callbacks", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "session", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "walletConnectOptions", void 0);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "isActive", false);
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "disconnect", /*#__PURE__*/function () {
      var _ref = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(sync) {
        var _this$callbacks, _this$callbacks$befor, _this$client, _this$client$disconne, _this$callbacks2, _this$callbacks2$afte;
        var _this$emitter, _this$logger;
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) switch (_context.prev = _context.next) {
            case 0:
              if (sync) {
                (_this$emitter = _this.emitter) === null || _this$emitter === void 0 ? void 0 : _this$emitter.emit('sync_disconnect', (0, _assertThisInitialized2["default"])(_this).chainName);
                (_this$logger = _this.logger) === null || _this$logger === void 0 ? void 0 : _this$logger.debug('[WALLET EVENT] Emit `sync_disconnect`');
              }
              _context.next = 3;
              return (_this$callbacks = _this.callbacks) === null || _this$callbacks === void 0 ? void 0 : (_this$callbacks$befor = _this$callbacks.beforeDisconnect) === null || _this$callbacks$befor === void 0 ? void 0 : _this$callbacks$befor.call(_this$callbacks);
            case 3:
              _this.reset();
              window.localStorage.removeItem('cosmos-kit@1:core//current-wallet');
              _context.next = 7;
              return (_this$client = _this.client) === null || _this$client === void 0 ? void 0 : (_this$client$disconne = _this$client.disconnect) === null || _this$client$disconne === void 0 ? void 0 : _this$client$disconne.call(_this$client);
            case 7:
              _context.next = 9;
              return (_this$callbacks2 = _this.callbacks) === null || _this$callbacks2 === void 0 ? void 0 : (_this$callbacks2$afte = _this$callbacks2.afterDisconnect) === null || _this$callbacks2$afte === void 0 ? void 0 : _this$callbacks2$afte.call(_this$callbacks2);
            case 9:
            case "end":
              return _context.stop();
          }
        }, _callee);
      }));
      return function (_x) {
        return _ref.apply(this, arguments);
      };
    }());
    (0, _defineProperty2["default"])((0, _assertThisInitialized2["default"])(_this), "connect", /*#__PURE__*/function () {
      var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2(sync) {
        var _this$callbacks3, _this$callbacks3$befo, _this$callbacks4, _this$callbacks4$afte;
        var _this$emitter2, _this$logger2, _this$emitter3, _this$logger3;
        return _regenerator["default"].wrap(function _callee2$(_context2) {
          while (1) switch (_context2.prev = _context2.next) {
            case 0:
              _context2.next = 2;
              return (_this$callbacks3 = _this.callbacks) === null || _this$callbacks3 === void 0 ? void 0 : (_this$callbacks3$befo = _this$callbacks3.beforeConnect) === null || _this$callbacks3$befo === void 0 ? void 0 : _this$callbacks3$befo.call(_this$callbacks3);
            case 2:
              if (!(_this.isMobile && _this.walletInfo.mobileDisabled)) {
                _context2.next = 5;
                break;
              }
              _this.setError('This wallet is not supported on mobile, please use desktop browsers.');
              return _context2.abrupt("return");
            case 5:
              if (sync) {
                (_this$emitter2 = _this.emitter) === null || _this$emitter2 === void 0 ? void 0 : _this$emitter2.emit('sync_connect', (0, _assertThisInitialized2["default"])(_this).chainName);
                (_this$logger2 = _this.logger) === null || _this$logger2 === void 0 ? void 0 : _this$logger2.debug('[WALLET EVENT] Emit `sync_connect`');
              }
              _context2.prev = 6;
              if (_this.client) {
                _context2.next = 17;
                break;
              }
              _this.setState(_types.State.Pending);
              _this.setMessage('InitClient');
              _context2.next = 12;
              return _this.initClient(_this.walletInfo.mode === 'wallet-connect' ? _this.walletConnectOptions : void 0);
            case 12:
              (_this$emitter3 = _this.emitter) === null || _this$emitter3 === void 0 ? void 0 : _this$emitter3.emit('broadcast_client', _this.client);
              (_this$logger3 = _this.logger) === null || _this$logger3 === void 0 ? void 0 : _this$logger3.debug('[WALLET EVENT] Emit `broadcast_client`');
              if (_this.client) {
                _context2.next = 17;
                break;
              }
              _this.setClientNotExist();
              return _context2.abrupt("return");
            case 17:
              _context2.next = 19;
              return _this.update();
            case 19:
              _context2.next = 24;
              break;
            case 21:
              _context2.prev = 21;
              _context2.t0 = _context2["catch"](6);
              _this.setError(_context2.t0);
            case 24:
              _context2.next = 26;
              return (_this$callbacks4 = _this.callbacks) === null || _this$callbacks4 === void 0 ? void 0 : (_this$callbacks4$afte = _this$callbacks4.afterConnect) === null || _this$callbacks4$afte === void 0 ? void 0 : _this$callbacks4$afte.call(_this$callbacks4);
            case 26:
            case "end":
              return _context2.stop();
          }
        }, _callee2, null, [[6, 21]]);
      }));
      return function (_x2) {
        return _ref2.apply(this, arguments);
      };
    }());
    _this._walletInfo = walletInfo;
    return _this;
  }
  (0, _createClass2["default"])(WalletBase, [{
    key: "appUrl",
    get: function get() {
      var _this$client2;
      return (_this$client2 = this.client) === null || _this$client2 === void 0 ? void 0 : _this$client2.appUrl;
    }
  }, {
    key: "qrUrl",
    get: function get() {
      var _this$client3;
      return (_this$client3 = this.client) === null || _this$client3 === void 0 ? void 0 : _this$client3.qrUrl;
    }
  }, {
    key: "activate",
    value: function activate() {
      this.isActive = true;
    }
  }, {
    key: "inactivate",
    value: function inactivate() {
      this.isActive = false;
    }
  }, {
    key: "client",
    get: function get() {
      var _this$clientMutable;
      return (_this$clientMutable = this.clientMutable) === null || _this$clientMutable === void 0 ? void 0 : _this$clientMutable.data;
    }
  }, {
    key: "initingClient",
    value: function initingClient() {
      this.clientMutable.state = _types.State.Pending;
    }
  }, {
    key: "initClientDone",
    value: function initClientDone(client) {
      this.clientMutable.data = client;
      this.clientMutable.state = _types.State.Done;
    }
  }, {
    key: "initClientError",
    value: function initClientError(error) {
      this.clientMutable.message = error === null || error === void 0 ? void 0 : error.message;
      this.clientMutable.state = _types.State.Error;
    }
  }, {
    key: "walletInfo",
    get: function get() {
      return this._walletInfo;
    }
  }, {
    key: "downloadInfo",
    get: function get() {
      var _this2 = this;
      var downloads = this.walletInfo.downloads || [];
      downloads = downloads.filter(function (d) {
        var _this2$env;
        return d.device === ((_this2$env = _this2.env) === null || _this2$env === void 0 ? void 0 : _this2$env.device) || !d.device;
      });
      if (downloads.length === 1) {
        return downloads[0];
      }
      downloads = downloads.filter(function (d) {
        var _this2$env2;
        return d.os === ((_this2$env2 = _this2.env) === null || _this2$env2 === void 0 ? void 0 : _this2$env2.os) || !d.os;
      });
      if (downloads.length === 1) {
        return downloads[0];
      }
      downloads = downloads.filter(function (d) {
        var _this2$env3;
        return d.browser === ((_this2$env3 = _this2.env) === null || _this2$env3 === void 0 ? void 0 : _this2$env3.browser) || !d.browser;
      });
      return downloads[0];
    }
  }, {
    key: "walletName",
    get: function get() {
      return this.walletInfo.name;
    }
  }, {
    key: "walletPrettyName",
    get: function get() {
      return this.walletInfo.prettyName;
    }
  }, {
    key: "rejectMessageSource",
    get: function get() {
      if (typeof this.walletInfo.rejectMessage === 'string') {
        return this.walletInfo.rejectMessage;
      } else {
        var _this$walletInfo$reje;
        return (_this$walletInfo$reje = this.walletInfo.rejectMessage) === null || _this$walletInfo$reje === void 0 ? void 0 : _this$walletInfo$reje.source;
      }
    }
  }, {
    key: "rejectMessageTarget",
    get: function get() {
      if (typeof this.walletInfo.rejectMessage === 'string') {
        return void 0;
      } else {
        var _this$walletInfo$reje2;
        return (_this$walletInfo$reje2 = this.walletInfo.rejectMessage) === null || _this$walletInfo$reje2 === void 0 ? void 0 : _this$walletInfo$reje2.target;
      }
    }
  }, {
    key: "rejectCode",
    get: function get() {
      return this.walletInfo.rejectCode;
    }
  }, {
    key: "rejectMatched",
    value: function rejectMatched(e) {
      return this.rejectMessageSource && e.message === this.rejectMessageSource || this.rejectCode && e.code === this.rejectCode;
    }
  }, {
    key: "updateCallbacks",
    value: function updateCallbacks(callbacks) {
      this.callbacks = _objectSpread(_objectSpread({}, this.callbacks), callbacks);
    }
  }, {
    key: "setClientNotExist",
    value: function setClientNotExist() {
      this.setState(_types.State.Error);
      this.setMessage(_utils.ClientNotExistError.message);
    }
  }, {
    key: "setRejected",
    value: function setRejected() {
      this.setState(_types.State.Error);
      this.setMessage(_utils.RejectedError.message);
    }
  }, {
    key: "setError",
    value: function setError(e) {
      this.setState(_types.State.Error);
      this.setMessage(typeof e === 'string' ? e : e === null || e === void 0 ? void 0 : e.message);
      if (typeof e !== 'string' && e !== null && e !== void 0 && e.stack) {
        var _this$logger4;
        (_this$logger4 = this.logger) === null || _this$logger4 === void 0 ? void 0 : _this$logger4.error(e.stack);
      }
    }
  }]);
  return WalletBase;
}(_state.StateBase);
exports.WalletBase = WalletBase;
