// @cosmology-ui/utils/main/cosmos/buttons/InstallWalletButton.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InstallWalletButtonBaseStyle = exports.InstallWalletButton = void 0;
var _react = require("@chakra-ui/react");
var _react2 = _interopRequireWildcard(require("react"));
var _index = require("../../index");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var InstallWalletButtonBaseStyle = function InstallWalletButtonBaseStyle(theme) {
  return {
    w: 'full',
    h: 'auto',
    py: 3,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    whiteSpace: 'break-spaces',
    fontWeight: 'medium',
    fontSize: 'md',
    border: '1px solid',
    borderColor: "install-wallet-button-border-color-".concat(theme),
    color: "install-wallet-button-text-color-".concat(theme),
    bg: "install-wallet-button-background-color-".concat(theme),
    boxShadow: "install-wallet-button-shadow-".concat(theme),
    _hover: {
      opacity: 0.8
    },
    _active: {
      opacity: 0.9
    },
    _focus: {
      outline: 'none'
    },
    _disabled: {
      opacity: 0.5,
      cursor: 'not-allowed',
      _hover: {
        opacity: 0.5
      },
      _active: {
        opacity: 0.5
      }
    },
    '>.install-wallet-button': {
      '>svg': {
        mr: 1,
        w: 'inherit'
      },
      '>img': {
        mr: 1.5,
        maxW: 3.5
      }
    }
  };
};
exports.InstallWalletButtonBaseStyle = InstallWalletButtonBaseStyle;
var InstallWalletButton = function InstallWalletButton(_ref) {
  var icon = _ref.icon,
    _ref$buttonText = _ref.buttonText,
    buttonText = _ref$buttonText === void 0 ? 'Install Wallet' : _ref$buttonText,
    _ref$disabled = _ref.disabled,
    disabled = _ref$disabled === void 0 ? false : _ref$disabled,
    className = _ref.className,
    styleProps = _ref.styleProps,
    onClick = _ref.onClick;
  var _useContext = (0, _react2.useContext)(_index.ThemeContext),
    theme = _useContext.theme;
  return /*#__PURE__*/_react2["default"].createElement(_react.Button, {
    className: className,
    variant: "unstyled",
    sx: styleProps ? styleProps : InstallWalletButtonBaseStyle(theme),
    isDisabled: disabled,
    onClick: onClick
  }, /*#__PURE__*/_react2["default"].createElement(_react.Center, {
    className: "install-wallet-button"
  }, typeof icon === 'string' ? /*#__PURE__*/_react2["default"].createElement(_react.Image, {
    src: icon,
    w: "full",
    h: "full",
    alt: "logo"
  }) : /*#__PURE__*/_react2["default"].createElement(_react.Icon, {
    as: icon,
    w: "full",
    h: "full"
  })), buttonText);
};
exports.InstallWalletButton = InstallWalletButton;
