// @cosmos-kit/cosmostation-mobile/main/wallet-connect/registry.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.cosmostationMobileInfo = void 0;
var cosmostationMobileInfo = {
  name: 'cosmostation-mobile',
  prettyName: 'Cosmostation Mobile',
  logo: 'https://raw.githubusercontent.com/cosmology-tech/cosmos-kit/main/public/images/logos/wallets/cosmostation.png',
  mode: 'wallet-connect',
  downloads: [{
    device: 'mobile',
    os: 'android',
    link: 'https://play.google.com/store/apps/details?id=wannabit.io.cosmostaion'
  }, {
    device: 'mobile',
    os: 'ios',
    link: 'https://apps.apple.com/kr/app/cosmostation/id1459830339'
  }, {
    link: 'https://cosmostation.io/wallet/#extension'
  }],
  mobileDisabled: false,
  walletconnect: {
    name: 'Cosmostation',
    projectId: 'feb6ff1fb426db18110f5a80c7adbde846d0a7e96b2bc53af4b73aaf32552bea',
    formatNativeUrl: function formatNativeUrl(appUrl, wcUri, name) {
      var plainAppUrl = appUrl.replaceAll('/', '').replaceAll(':', '');
      // const encodedWcUrl = encodeURIComponent(wcUri);
      return "".concat(plainAppUrl, "://wc?").concat(wcUri);
    }
  }
};
exports.cosmostationMobileInfo = cosmostationMobileInfo;
