// @cosmos-kit/react/main/modal/theme.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addSemanticTokens = addSemanticTokens;
exports.noCssResetTheme = exports.defaultTheme = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("@chakra-ui/react");
var _react2 = require("@cosmology-ui/react");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
/* ====================================================== */
/*       NOTICE: Chakra resets global CSS by default      */
/* ====================================================== */

var config = {
  initialColorMode: 'light',
  useSystemColorMode: true
};
var colors = {
  primary: {
    '50': '#e5e7f9',
    '100': '#bec4ef',
    '200': '#929ce4',
    '300': '#6674d9',
    '400': '#4657d1',
    '500': '#2539c9',
    '600': '#2133c3',
    '700': '#1b2cbc',
    '800': '#1624b5',
    '900': '#0d17a9'
  }
};

/* ----- this theme is for without Chakra CSS reset ----- */
var themeWithoutReset = {
  styles: {
    global: {
      html: {
        fontFamily: '',
        transitionProperty: '',
        transitionDuration: '',
        lineHeight: 'normal',
        WebkitTextSizeAdjust: '',
        WebkitFontSmoothing: 'auto',
        textRendering: '',
        MozOsxFontSmoothing: '',
        touchAction: ''
      },
      body: {
        fontFamily: '',
        color: '',
        bg: '',
        position: 'relative',
        minHeight: '100%',
        fontFeatureSettings: ''
      },
      '*::placeholder': {
        color: ''
      },
      '*, *::before, *::after': {
        borderWidth: '',
        borderStyle: '',
        boxSizing: ''
      },
      '*, *::before, &::after': {
        borderColor: '',
        wordWrap: ''
      },
      main: {
        display: ''
      },
      hr: {
        borderTopWidth: '',
        boxSizing: '',
        height: '',
        overflow: ''
      },
      'pre, code, kbd, samp': {
        fontFamily: '',
        fontSize: ''
      },
      a: {
        backgroundColor: '',
        color: '',
        textDecoration: ''
      },
      'abbr[title]': {
        borderBottom: '',
        WebkitTextDecoration: '',
        textDecoration: ''
      },
      'b, strong': {
        fontWeight: ''
      },
      small: {
        fontSize: ''
      },
      'sub, sup': {
        fontSize: '',
        lineHeight: '',
        position: '',
        verticalAlign: ''
      },
      sub: {
        bottom: ''
      },
      sup: {
        top: ''
      },
      img: {
        borderStyle: ''
      },
      'button, input, optgroup, select,textarea': {
        fontFamily: '',
        fontSize: '',
        lineHeight: '',
        margin: ''
      },
      'button, input': {
        overflow: ''
      },
      'button, select': {
        textTransform: ''
      },
      'button::-moz-focus-inner, [type="button"]::-moz-focus-inner, [type="reset"]::-moz-focus-inner, [type="submit"]::-moz-focus-inner': {
        borderStyle: '',
        padding: ''
      },
      fieldset: {
        margin: '',
        padding: ''
      },
      legend: {
        boxSizing: '',
        color: '',
        display: '',
        maxWidth: '',
        padding: '',
        whiteSpace: ''
      },
      progress: {
        verticalAlign: ''
      },
      textarea: {
        overflow: '',
        resize: ''
      },
      '[type="checkbox"], [type="radio"]': {
        boxSizing: '',
        padding: ''
      },
      '[type="number"]::-webkit-inner-spin-button, [type="number"]::-webkit-outer-spin-button': {
        WebkitAppearance: ''
      },
      'input[type="number"]': {
        MozAppearance: ''
      },
      '[type="search"]': {
        '-webkit-appearance': '',
        outlineOffset: ''
      },
      '[type="search"]::-webkit-search-decoration': {
        WebkitAppearance: ''
      },
      '::-webkit-file-upload-button': {
        WebkitAppearance: '',
        font: ''
      },
      details: {
        display: ''
      },
      summary: {
        display: ''
      },
      template: {
        display: ''
      },
      '[hidden]': {
        display: ''
      },
      'body, blockquote, dl, dd, h1, h2, h3, h4, h5, h6, hr, figure, p, pre': {
        margin: ''
      },
      button: {
        background: '',
        padding: ''
      },
      'button,[role="button"]': {
        cursor: ''
      },
      'button::-moz-focus-inner': {
        border: ''
      },
      table: {
        borderCollapse: ''
      },
      'ol, ul': {
        margin: '',
        padding: ''
      },
      'h1, h2, h3, h4, h5, h6': {
        fontSize: '',
        fontWeight: ''
      },
      'button, input, optgroup, select, textarea': {
        padding: '',
        lineHeight: '',
        color: ''
      },
      'img, svg, video, canvas, audio, iframe, embed, object': {
        display: ''
      },
      'img, video': {
        maxWidth: '',
        height: ''
      },
      '[data-js-focus-visible] :focus:not([data-focus-visible-added]):not([data-focus-visible-disabled])': {
        outline: '',
        boxShadow: ''
      },
      'select::-ms-expand': {
        display: ''
      }
    }
  },
  colors: colors,
  config: config
};

/* ------- this theme is for with Chakra CSS reset ------ */
var themeWithReset = {
  fonts: {
    body: 'Inter, system-ui, sans-serif',
    heading: 'Work Sans, system-ui, sans-serif'
  },
  colors: colors,
  config: config
};
function addSemanticTokens(theme) {
  var _semanticTokens, _semanticTokens2;
  var colors = _objectSpread(_objectSpread({}, _react2.semanticTokens.semanticTokens.colors), theme === null || theme === void 0 ? void 0 : (_semanticTokens = theme.semanticTokens) === null || _semanticTokens === void 0 ? void 0 : _semanticTokens.colors);
  var shadows = _objectSpread(_objectSpread({}, _react2.semanticTokens.semanticTokens.shadows), theme === null || theme === void 0 ? void 0 : (_semanticTokens2 = theme.semanticTokens) === null || _semanticTokens2 === void 0 ? void 0 : _semanticTokens2.shadows);
  var mergedSemanticTokens = {
    semanticTokens: _objectSpread(_objectSpread({}, theme === null || theme === void 0 ? void 0 : theme.semanticTokens), {}, {
      colors: colors,
      shadows: shadows
    })
  };
  return _objectSpread(_objectSpread({}, theme), mergedSemanticTokens);
}
var defaultTheme = addSemanticTokens((0, _react.extendTheme)(themeWithReset));

// noCssResetTheme removed most global css reset made by chakra
exports.defaultTheme = defaultTheme;
var noCssResetTheme = addSemanticTokens((0, _react.extendTheme)(themeWithoutReset));
exports.noCssResetTheme = noCssResetTheme;
