// @cosmos-kit/leap-extension/main/extension/client.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LeapClient = void 0;
var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));
var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var LeapClient = /*#__PURE__*/function () {
  function LeapClient(client) {
    (0, _classCallCheck2["default"])(this, LeapClient);
    (0, _defineProperty2["default"])(this, "client", void 0);
    this.client = client;
  }
  (0, _createClass2["default"])(LeapClient, [{
    key: "enable",
    value: function () {
      var _enable = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(chainIds) {
        return _regenerator["default"].wrap(function _callee$(_context) {
          while (1) switch (_context.prev = _context.next) {
            case 0:
              _context.next = 2;
              return this.client.enable(chainIds);
            case 2:
            case "end":
              return _context.stop();
          }
        }, _callee, this);
      }));
      function enable(_x) {
        return _enable.apply(this, arguments);
      }
      return enable;
    }()
  }, {
    key: "disconnect",
    value: function () {
      var _disconnect = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
        return _regenerator["default"].wrap(function _callee2$(_context2) {
          while (1) switch (_context2.prev = _context2.next) {
            case 0:
              _context2.next = 2;
              return this.client.disconnect();
            case 2:
            case "end":
              return _context2.stop();
          }
        }, _callee2, this);
      }));
      function disconnect() {
        return _disconnect.apply(this, arguments);
      }
      return disconnect;
    }()
  }, {
    key: "getSimpleAccount",
    value: function () {
      var _getSimpleAccount = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(chainId) {
        var _yield$this$getAccoun, address, username;
        return _regenerator["default"].wrap(function _callee3$(_context3) {
          while (1) switch (_context3.prev = _context3.next) {
            case 0:
              _context3.next = 2;
              return this.getAccount(chainId);
            case 2:
              _yield$this$getAccoun = _context3.sent;
              address = _yield$this$getAccoun.address;
              username = _yield$this$getAccoun.username;
              return _context3.abrupt("return", {
                namespace: 'cosmos',
                chainId: chainId,
                address: address,
                username: username
              });
            case 6:
            case "end":
              return _context3.stop();
          }
        }, _callee3, this);
      }));
      function getSimpleAccount(_x2) {
        return _getSimpleAccount.apply(this, arguments);
      }
      return getSimpleAccount;
    }()
  }, {
    key: "getAccount",
    value: function () {
      var _getAccount = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(chainId) {
        var key;
        return _regenerator["default"].wrap(function _callee4$(_context4) {
          while (1) switch (_context4.prev = _context4.next) {
            case 0:
              _context4.next = 2;
              return this.client.getKey(chainId);
            case 2:
              key = _context4.sent;
              return _context4.abrupt("return", {
                username: key.name,
                address: key.bech32Address,
                algo: key.algo,
                pubkey: key.pubKey
              });
            case 4:
            case "end":
              return _context4.stop();
          }
        }, _callee4, this);
      }));
      function getAccount(_x3) {
        return _getAccount.apply(this, arguments);
      }
      return getAccount;
    }()
  }, {
    key: "getOfflineSigner",
    value: function getOfflineSigner(chainId, preferredSignType) {
      switch (preferredSignType) {
        case 'amino':
          return this.getOfflineSignerAmino(chainId);
        case 'direct':
          return this.getOfflineSignerDirect(chainId);
        default:
          return this.getOfflineSignerAmino(chainId);
      }
      // return this.client.getOfflineSignerAuto(chainId);
    }
  }, {
    key: "getOfflineSignerAmino",
    value: function getOfflineSignerAmino(chainId) {
      return this.client.getOfflineSignerOnlyAmino(chainId);
    }
  }, {
    key: "getOfflineSignerDirect",
    value: function getOfflineSignerDirect(chainId) {
      return this.client.getOfflineSigner(chainId);
    }
  }, {
    key: "signAmino",
    value: function () {
      var _signAmino = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(chainId, signer, signDoc, signOptions) {
        return _regenerator["default"].wrap(function _callee5$(_context5) {
          while (1) switch (_context5.prev = _context5.next) {
            case 0:
              _context5.next = 2;
              return this.client.signAmino(chainId, signer, signDoc, signOptions);
            case 2:
              return _context5.abrupt("return", _context5.sent);
            case 3:
            case "end":
              return _context5.stop();
          }
        }, _callee5, this);
      }));
      function signAmino(_x4, _x5, _x6, _x7) {
        return _signAmino.apply(this, arguments);
      }
      return signAmino;
    }()
  }, {
    key: "signDirect",
    value: function () {
      var _signDirect = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6(chainId, signer, signDoc, signOptions) {
        return _regenerator["default"].wrap(function _callee6$(_context6) {
          while (1) switch (_context6.prev = _context6.next) {
            case 0:
              _context6.next = 2;
              return this.client.signDirect(chainId, signer, signDoc, signOptions);
            case 2:
              return _context6.abrupt("return", _context6.sent);
            case 3:
            case "end":
              return _context6.stop();
          }
        }, _callee6, this);
      }));
      function signDirect(_x8, _x9, _x10, _x11) {
        return _signDirect.apply(this, arguments);
      }
      return signDirect;
    }()
  }, {
    key: "sendTx",
    value: function () {
      var _sendTx = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee7(chainId, tx, mode) {
        return _regenerator["default"].wrap(function _callee7$(_context7) {
          while (1) switch (_context7.prev = _context7.next) {
            case 0:
              _context7.next = 2;
              return this.client.sendTx(chainId, tx, mode);
            case 2:
              return _context7.abrupt("return", _context7.sent);
            case 3:
            case "end":
              return _context7.stop();
          }
        }, _callee7, this);
      }));
      function sendTx(_x12, _x13, _x14) {
        return _sendTx.apply(this, arguments);
      }
      return sendTx;
    }()
  }]);
  return LeapClient;
}();
exports.LeapClient = LeapClient;
