// @cosmos-kit/react-lite/main/provider.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.walletContext = exports.ChainProvider = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _core = require("@cosmos-kit/core");
var _react = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var walletContext = /*#__PURE__*/(0, _react.createContext)(null);
exports.walletContext = walletContext;
var ChainProvider = function ChainProvider(_ref) {
  var chains = _ref.chains,
    assetLists = _ref.assetLists,
    wallets = _ref.wallets,
    ProvidedWalletModal = _ref.walletModal,
    _ref$throwErrors = _ref.throwErrors,
    throwErrors = _ref$throwErrors === void 0 ? false : _ref$throwErrors,
    _ref$defaultNameServi = _ref.defaultNameService,
    defaultNameService = _ref$defaultNameServi === void 0 ? 'icns' : _ref$defaultNameServi,
    walletConnectOptions = _ref.walletConnectOptions,
    signerOptions = _ref.signerOptions,
    endpointOptions = _ref.endpointOptions,
    sessionOptions = _ref.sessionOptions,
    _ref$logLevel = _ref.logLevel,
    logLevel = _ref$logLevel === void 0 ? 'WARN' : _ref$logLevel,
    children = _ref.children;
  var logger = (0, _react.useMemo)(function () {
    return new _core.Logger(logLevel);
  }, []);
  var walletManager = (0, _react.useMemo)(function () {
    return new _core.WalletManager(chains, assetLists, wallets, logger, throwErrors, defaultNameService, walletConnectOptions, signerOptions, endpointOptions, sessionOptions);
  }, []);
  var _useState = (0, _react.useState)(false),
    _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
    isViewOpen = _useState2[0],
    setViewOpen = _useState2[1];
  var _useState3 = (0, _react.useState)(),
    _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
    viewWalletRepo = _useState4[0],
    setViewWalletRepo = _useState4[1];
  var _useState5 = (0, _react.useState)(),
    _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
    setData = _useState6[1];
  var _useState7 = (0, _react.useState)(_core.State.Init),
    _useState8 = (0, _slicedToArray2["default"])(_useState7, 2),
    setState = _useState8[1];
  var _useState9 = (0, _react.useState)(),
    _useState10 = (0, _slicedToArray2["default"])(_useState9, 2),
    setMsg = _useState10[1];
  var _useState11 = (0, _react.useState)(_core.State.Init),
    _useState12 = (0, _slicedToArray2["default"])(_useState11, 2),
    setClientState = _useState12[1];
  walletManager.setActions({
    viewOpen: setViewOpen,
    viewWalletRepo: setViewWalletRepo,
    data: setData,
    state: setState,
    message: setMsg
  });
  walletManager.walletRepos.forEach(function (wr) {
    wr.setActions({
      viewOpen: setViewOpen,
      viewWalletRepo: setViewWalletRepo
    });
    wr.wallets.forEach(function (w) {
      w.setActions({
        data: setData,
        state: setState,
        message: setMsg
      });
    });
  });
  walletManager.mainWallets.forEach(function (w) {
    w.setActions({
      data: setData,
      state: setState,
      message: setMsg,
      clientState: setClientState
    });
  });
  (0, _react.useEffect)(function () {
    walletManager.onMounted();
    return function () {
      setViewOpen(false);
      walletManager.onUnmounted();
    };
  }, []);
  return /*#__PURE__*/_react["default"].createElement(walletContext.Provider, {
    value: {
      walletManager: walletManager,
      modalProvided: Boolean(ProvidedWalletModal)
    }
  }, ProvidedWalletModal && /*#__PURE__*/_react["default"].createElement(ProvidedWalletModal, {
    isOpen: isViewOpen,
    setOpen: setViewOpen,
    walletRepo: viewWalletRepo
  }), children);
};
exports.ChainProvider = ChainProvider;
