// @cosmos-kit/react/module/modal/components/views/Connecting.js
import { LogoStatus, SimpleDisplayModalContent, SimpleModalHead, SimpleModalView } from '@cosmology-ui/react';
import React from 'react';
export const ConnectingView = ({
  onClose,
  onReturn,
  wallet
}) => {
  const {
    walletInfo: {
      prettyName,
      logo,
      mode
    },
    message
  } = wallet;
  let title = 'Requesting Connection';
  let desc = mode === 'wallet-connect' ? `Approve ${prettyName} connection request on your mobile.` : `Open the ${prettyName} browser extension to connect your wallet.`;
  if (message === 'InitClient') {
    title = 'Initializing Wallet Client';
    desc = '';
  }
  const modalHead = /*#__PURE__*/React.createElement(SimpleModalHead, {
    title: prettyName,
    backButton: true,
    onClose: onClose,
    onBack: onReturn
  });
  const modalContent = /*#__PURE__*/React.createElement(SimpleDisplayModalContent, {
    status: LogoStatus.Loading,
    logo: logo,
    contentHeader: title,
    contentDesc: desc
  });
  return /*#__PURE__*/React.createElement(SimpleModalView, {
    modalHead: modalHead,
    modalContent: modalContent
  });
};
