// @cosmos-kit/react/main/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  walletContext: true,
  useChain: true,
  useChainWallet: true,
  useManager: true,
  useWallet: true,
  useWalletClient: true,
  useNameService: true
};
Object.defineProperty(exports, "useChain", {
  enumerable: true,
  get: function get() {
    return _reactLite.useChain;
  }
});
Object.defineProperty(exports, "useChainWallet", {
  enumerable: true,
  get: function get() {
    return _reactLite.useChainWallet;
  }
});
Object.defineProperty(exports, "useManager", {
  enumerable: true,
  get: function get() {
    return _reactLite.useManager;
  }
});
Object.defineProperty(exports, "useNameService", {
  enumerable: true,
  get: function get() {
    return _reactLite.useNameService;
  }
});
Object.defineProperty(exports, "useWallet", {
  enumerable: true,
  get: function get() {
    return _reactLite.useWallet;
  }
});
Object.defineProperty(exports, "useWalletClient", {
  enumerable: true,
  get: function get() {
    return _reactLite.useWalletClient;
  }
});
Object.defineProperty(exports, "walletContext", {
  enumerable: true,
  get: function get() {
    return _reactLite.walletContext;
  }
});
var _reactLite = require("@cosmos-kit/react-lite");
var _modal = require("./modal");
Object.keys(_modal).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _modal[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _modal[key];
    }
  });
});
var _provider = require("./provider");
Object.keys(_provider).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _provider[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _provider[key];
    }
  });
});
var _hooks = require("./hooks");
Object.keys(_hooks).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _hooks[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _hooks[key];
    }
  });
});
