// @cosmos-kit/react/main/modal/components/views/Connecting.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ConnectingView = void 0;
var _react = require("@cosmology-ui/react");
var _react2 = _interopRequireDefault(require("react"));
var ConnectingView = function ConnectingView(_ref) {
  var onClose = _ref.onClose,
    onReturn = _ref.onReturn,
    wallet = _ref.wallet;
  var _wallet$walletInfo = wallet.walletInfo,
    prettyName = _wallet$walletInfo.prettyName,
    logo = _wallet$walletInfo.logo,
    mode = _wallet$walletInfo.mode,
    message = wallet.message;
  var title = 'Requesting Connection';
  var desc = mode === 'wallet-connect' ? "Approve ".concat(prettyName, " connection request on your mobile.") : "Open the ".concat(prettyName, " browser extension to connect your wallet.");
  if (message === 'InitClient') {
    title = 'Initializing Wallet Client';
    desc = '';
  }
  var modalHead = /*#__PURE__*/_react2["default"].createElement(_react.SimpleModalHead, {
    title: prettyName,
    backButton: true,
    onClose: onClose,
    onBack: onReturn
  });
  var modalContent = /*#__PURE__*/_react2["default"].createElement(_react.SimpleDisplayModalContent, {
    status: _react.LogoStatus.Loading,
    logo: logo,
    contentHeader: title,
    contentDesc: desc
  });
  return /*#__PURE__*/_react2["default"].createElement(_react.SimpleModalView, {
    modalHead: modalHead,
    modalContent: modalContent
  });
};
exports.ConnectingView = ConnectingView;
