// @cosmos-kit/react/main/modal/components/views/Error.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ErrorView = void 0;
var _react = require("@chakra-ui/react");
var _react2 = require("@cosmology-ui/react");
var _react3 = _interopRequireDefault(require("react"));
var ErrorView = function ErrorView(_ref) {
  var onClose = _ref.onClose,
    onReturn = _ref.onReturn,
    wallet = _ref.wallet;
  var _wallet$walletInfo = wallet.walletInfo,
    prettyName = _wallet$walletInfo.prettyName,
    logo = _wallet$walletInfo.logo,
    message = wallet.message;
  var modalHead = /*#__PURE__*/_react3["default"].createElement(_react2.SimpleModalHead, {
    title: prettyName,
    backButton: true,
    onClose: onClose,
    onBack: onReturn
  });
  var modalContent = /*#__PURE__*/_react3["default"].createElement(_react2.SimpleDisplayModalContent, {
    status: _react2.LogoStatus.Error,
    logo: logo,
    contentHeader: 'Oops! Something wrong...',
    contentDesc: message,
    bottomButton: /*#__PURE__*/_react3["default"].createElement(_react.Box, {
      px: 6
    }, /*#__PURE__*/_react3["default"].createElement(_react2.ConnectWalletButton, {
      buttonText: 'Change Wallet',
      onClick: onReturn
    }))
  });
  return /*#__PURE__*/_react3["default"].createElement(_react2.SimpleModalView, {
    modalHead: modalHead,
    modalContent: modalContent
  });
};
exports.ErrorView = ErrorView;
