// @cosmos-kit/react/main/modal/components/views/NotExist.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NotExistView = void 0;
var _react = require("@cosmology-ui/react");
var _react2 = _interopRequireWildcard(require("react"));
var _go = require("react-icons/go");
var _ri = require("react-icons/ri");
var _fa = require("react-icons/fa");
var _gr = require("react-icons/gr");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var NotExistView = function NotExistView(_ref) {
  var onClose = _ref.onClose,
    onReturn = _ref.onReturn,
    wallet = _ref.wallet;
  var _wallet$walletInfo = wallet.walletInfo,
    prettyName = _wallet$walletInfo.prettyName,
    logo = _wallet$walletInfo.logo,
    downloadInfo = wallet.downloadInfo;
  var onInstall = (0, _react2.useCallback)(function () {
    window.open(downloadInfo === null || downloadInfo === void 0 ? void 0 : downloadInfo.link, '_blank');
  }, [downloadInfo]);
  var icon = (0, _react2.useMemo)(function () {
    if ((downloadInfo === null || downloadInfo === void 0 ? void 0 : downloadInfo.browser) === 'chrome') return _ri.RiChromeFill;
    if ((downloadInfo === null || downloadInfo === void 0 ? void 0 : downloadInfo.browser) === 'firefox') return _gr.GrFirefox;
    if ((downloadInfo === null || downloadInfo === void 0 ? void 0 : downloadInfo.os) === 'android') return _fa.FaAndroid;
    if ((downloadInfo === null || downloadInfo === void 0 ? void 0 : downloadInfo.os) === 'ios') return _ri.RiAppStoreFill;
    return _go.GoDesktopDownload;
  }, [downloadInfo]);
  var modalHead = /*#__PURE__*/_react2["default"].createElement(_react.SimpleModalHead, {
    title: prettyName,
    backButton: true,
    onClose: onClose,
    onBack: onReturn
  });
  var modalContent = /*#__PURE__*/_react2["default"].createElement(_react.SimpleDisplayModalContent, {
    status: _react.LogoStatus.Error,
    logo: logo,
    contentHeader: "".concat(prettyName, " Not Installed"),
    contentDesc: onInstall ? "If ".concat(prettyName.toLowerCase(), " is installed on your device, please refresh this page or follow the browser instruction.") : "Download link not provided. Try searching it or consulting the developer team.",
    bottomButton: /*#__PURE__*/_react2["default"].createElement(_react.InstallWalletButton, {
      icon: icon,
      buttonText: "Install ".concat(prettyName),
      onClick: onInstall,
      disabled: !(downloadInfo !== null && downloadInfo !== void 0 && downloadInfo.link)
    })
  });
  return /*#__PURE__*/_react2["default"].createElement(_react.SimpleModalView, {
    modalHead: modalHead,
    modalContent: modalContent
  });
};
exports.NotExistView = NotExistView;
