// @cosmos-kit/react/main/modal/components/views/Rejected.js
"use strict";

var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RejectedView = void 0;
var _react = require("@chakra-ui/react");
var _react2 = require("@cosmology-ui/react");
var _react3 = _interopRequireWildcard(require("react"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var RejectedView = function RejectedView(_ref) {
  var onClose = _ref.onClose,
    onReturn = _ref.onReturn,
    wallet = _ref.wallet;
  var _wallet$walletInfo = wallet.walletInfo,
    prettyName = _wallet$walletInfo.prettyName,
    logo = _wallet$walletInfo.logo;
  var onReconnect = (0, _react3.useCallback)(function () {
    wallet.connect(false);
  }, [wallet]);
  var modalHead = /*#__PURE__*/_react3["default"].createElement(_react2.SimpleModalHead, {
    title: prettyName,
    backButton: true,
    onClose: onClose,
    onBack: onReturn
  });
  var modalContent = /*#__PURE__*/_react3["default"].createElement(_react2.SimpleDisplayModalContent, {
    status: _react2.LogoStatus.Error,
    logo: logo,
    contentHeader: 'Request Rejected',
    contentDesc: wallet.rejectMessageTarget || 'Connection permission is denied.',
    bottomButton: /*#__PURE__*/_react3["default"].createElement(_react.Box, {
      px: 6
    }, /*#__PURE__*/_react3["default"].createElement(_react2.ConnectWalletButton, {
      buttonText: 'Reconnect',
      onClick: onReconnect
    }))
  });
  return /*#__PURE__*/_react3["default"].createElement(_react2.SimpleModalView, {
    modalHead: modalHead,
    modalContent: modalContent
  });
};
exports.RejectedView = RejectedView;
