// @cosmos-kit/core/main/utils/convert.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.convertChain = convertChain;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _endpoint = require("./endpoint");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function convertChain(chain, assetLists, signerOptions, preferredEndpoints, isLazy, logger) {
  var _signerOptions$starga, _signerOptions$signin, _signerOptions$signin2, _signerOptions$prefer;
  var assetList = assetLists.find(function (list) {
    return list.chain_name === chain.chain_name;
  });
  return {
    name: chain.chain_name,
    chain: chain,
    assetList: assetList,
    clientOptions: {
      stargate: signerOptions === null || signerOptions === void 0 ? void 0 : (_signerOptions$starga = signerOptions.stargate) === null || _signerOptions$starga === void 0 ? void 0 : _signerOptions$starga.call(signerOptions, chain),
      signingStargate: signerOptions === null || signerOptions === void 0 ? void 0 : (_signerOptions$signin = signerOptions.signingStargate) === null || _signerOptions$signin === void 0 ? void 0 : _signerOptions$signin.call(signerOptions, chain),
      signingCosmwasm: signerOptions === null || signerOptions === void 0 ? void 0 : (_signerOptions$signin2 = signerOptions.signingCosmwasm) === null || _signerOptions$signin2 === void 0 ? void 0 : _signerOptions$signin2.call(signerOptions, chain),
      preferredSignType: (signerOptions === null || signerOptions === void 0 ? void 0 : (_signerOptions$prefer = signerOptions.preferredSignType) === null || _signerOptions$prefer === void 0 ? void 0 : _signerOptions$prefer.call(signerOptions, chain)) || 'amino'
    },
    preferredEndpoints: _objectSpread(_objectSpread({}, preferredEndpoints), {}, {
      isLazy: (0, _endpoint.getIsLazy)(isLazy, preferredEndpoints === null || preferredEndpoints === void 0 ? void 0 : preferredEndpoints.isLazy, void 0, void 0, logger)
    })
  };
}
