// stargazejs/main/codegen/stargaze/claim/v1beta1/query.rpc.Query.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createRpcQueryExtension = exports.QueryClientImpl = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _m0 = _interopRequireWildcard(require("protobufjs/minimal"));
var _stargate = require("@cosmjs/stargate");
var _query = require("./query");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var QueryClientImpl = /*#__PURE__*/function () {
  function QueryClientImpl(rpc) {
    (0, _classCallCheck2["default"])(this, QueryClientImpl);
    (0, _defineProperty2["default"])(this, "rpc", void 0);
    this.rpc = rpc;
    this.moduleAccountBalance = this.moduleAccountBalance.bind(this);
    this.params = this.params.bind(this);
    this.claimRecord = this.claimRecord.bind(this);
    this.claimableForAction = this.claimableForAction.bind(this);
    this.totalClaimable = this.totalClaimable.bind(this);
  }
  (0, _createClass2["default"])(QueryClientImpl, [{
    key: "moduleAccountBalance",
    value: function moduleAccountBalance() {
      var request = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      var data = _query.QueryModuleAccountBalanceRequest.encode(request).finish();
      var promise = this.rpc.request("publicawesome.stargaze.claim.v1beta1.Query", "ModuleAccountBalance", data);
      return promise.then(function (data) {
        return _query.QueryModuleAccountBalanceResponse.decode(new _m0.Reader(data));
      });
    }
  }, {
    key: "params",
    value: function params() {
      var request = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      var data = _query.QueryParamsRequest.encode(request).finish();
      var promise = this.rpc.request("publicawesome.stargaze.claim.v1beta1.Query", "Params", data);
      return promise.then(function (data) {
        return _query.QueryParamsResponse.decode(new _m0.Reader(data));
      });
    }
  }, {
    key: "claimRecord",
    value: function claimRecord(request) {
      var data = _query.QueryClaimRecordRequest.encode(request).finish();
      var promise = this.rpc.request("publicawesome.stargaze.claim.v1beta1.Query", "ClaimRecord", data);
      return promise.then(function (data) {
        return _query.QueryClaimRecordResponse.decode(new _m0.Reader(data));
      });
    }
  }, {
    key: "claimableForAction",
    value: function claimableForAction(request) {
      var data = _query.QueryClaimableForActionRequest.encode(request).finish();
      var promise = this.rpc.request("publicawesome.stargaze.claim.v1beta1.Query", "ClaimableForAction", data);
      return promise.then(function (data) {
        return _query.QueryClaimableForActionResponse.decode(new _m0.Reader(data));
      });
    }
  }, {
    key: "totalClaimable",
    value: function totalClaimable(request) {
      var data = _query.QueryTotalClaimableRequest.encode(request).finish();
      var promise = this.rpc.request("publicawesome.stargaze.claim.v1beta1.Query", "TotalClaimable", data);
      return promise.then(function (data) {
        return _query.QueryTotalClaimableResponse.decode(new _m0.Reader(data));
      });
    }
  }]);
  return QueryClientImpl;
}();
exports.QueryClientImpl = QueryClientImpl;
var createRpcQueryExtension = function createRpcQueryExtension(base) {
  var rpc = (0, _stargate.createProtobufRpcClient)(base);
  var queryService = new QueryClientImpl(rpc);
  return {
    moduleAccountBalance: function moduleAccountBalance(request) {
      return queryService.moduleAccountBalance(request);
    },
    params: function params(request) {
      return queryService.params(request);
    },
    claimRecord: function claimRecord(request) {
      return queryService.claimRecord(request);
    },
    claimableForAction: function claimableForAction(request) {
      return queryService.claimableForAction(request);
    },
    totalClaimable: function totalClaimable(request) {
      return queryService.totalClaimable(request);
    }
  };
};
exports.createRpcQueryExtension = createRpcQueryExtension;
