// stargazejs/main/codegen/stargaze/claim/v1beta1/tx.amino.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AminoConverter = void 0;
var _claim_record = require("./claim_record");
var AminoConverter = {
  "/publicawesome.stargaze.claim.v1beta1.MsgInitialClaim": {
    aminoType: "/publicawesome.stargaze.claim.v1beta1.MsgInitialClaim",
    toAmino: function toAmino(_ref) {
      var sender = _ref.sender;
      return {
        sender: sender
      };
    },
    fromAmino: function fromAmino(_ref2) {
      var sender = _ref2.sender;
      return {
        sender: sender
      };
    }
  },
  "/publicawesome.stargaze.claim.v1beta1.MsgClaimFor": {
    aminoType: "/publicawesome.stargaze.claim.v1beta1.MsgClaimFor",
    toAmino: function toAmino(_ref3) {
      var sender = _ref3.sender,
        address = _ref3.address,
        action = _ref3.action;
      return {
        sender: sender,
        address: address,
        action: action
      };
    },
    fromAmino: function fromAmino(_ref4) {
      var sender = _ref4.sender,
        address = _ref4.address,
        action = _ref4.action;
      return {
        sender: sender,
        address: address,
        action: (0, _claim_record.actionFromJSON)(action)
      };
    }
  }
};
exports.AminoConverter = AminoConverter;
