// stargazejs/main/codegen/stargaze/claim/v1beta1/tx.registry.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.registry = exports.load = exports.MessageComposer = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _tx = require("./tx");
var registry = [["/publicawesome.stargaze.claim.v1beta1.MsgInitialClaim", _tx.MsgInitialClaim], ["/publicawesome.stargaze.claim.v1beta1.MsgClaimFor", _tx.MsgClaimFor]];
exports.registry = registry;
var load = function load(protoRegistry) {
  registry.forEach(function (_ref) {
    var _ref2 = (0, _slicedToArray2["default"])(_ref, 2),
      typeUrl = _ref2[0],
      mod = _ref2[1];
    protoRegistry.register(typeUrl, mod);
  });
};
exports.load = load;
var MessageComposer = {
  encoded: {
    initialClaim: function initialClaim(value) {
      return {
        typeUrl: "/publicawesome.stargaze.claim.v1beta1.MsgInitialClaim",
        value: _tx.MsgInitialClaim.encode(value).finish()
      };
    },
    claimFor: function claimFor(value) {
      return {
        typeUrl: "/publicawesome.stargaze.claim.v1beta1.MsgClaimFor",
        value: _tx.MsgClaimFor.encode(value).finish()
      };
    }
  },
  withTypeUrl: {
    initialClaim: function initialClaim(value) {
      return {
        typeUrl: "/publicawesome.stargaze.claim.v1beta1.MsgInitialClaim",
        value: value
      };
    },
    claimFor: function claimFor(value) {
      return {
        typeUrl: "/publicawesome.stargaze.claim.v1beta1.MsgClaimFor",
        value: value
      };
    }
  },
  fromPartial: {
    initialClaim: function initialClaim(value) {
      return {
        typeUrl: "/publicawesome.stargaze.claim.v1beta1.MsgInitialClaim",
        value: _tx.MsgInitialClaim.fromPartial(value)
      };
    },
    claimFor: function claimFor(value) {
      return {
        typeUrl: "/publicawesome.stargaze.claim.v1beta1.MsgClaimFor",
        value: _tx.MsgClaimFor.fromPartial(value)
      };
    }
  }
};
exports.MessageComposer = MessageComposer;
